<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="DongyueXie" />

<meta name="date" content="2020-08-16" />

<title>cell size</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">deconference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="about.html">Data</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/DongyueXie/deconference">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">cell size</h1>
<h4 class="author"><em>DongyueXie</em></h4>
<h4 class="date"><em>2020-08-16</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>deconference/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200403code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200403)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200403code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200403)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/deconference/data/baron-human.rds
</td>
<td style="text-align:left;">
data/baron-human.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetreeea758b6b29abf5690e9a2a4e465bd69915b150c4targetblankea758b6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DongyueXie/deconference/tree/ea758b6b29abf5690e9a2a4e465bd69915b150c4" target="_blank">ea758b6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetreeea758b6b29abf5690e9a2a4e465bd69915b150c4targetblankea758b6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DongyueXie/deconference/tree/ea758b6b29abf5690e9a2a4e465bd69915b150c4" target="_blank">ea758b6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Xin_universal.RData
    Untracked:  Xin_universal_hc3.RData
    Untracked:  analysis/meta_mode.Rmd
    Untracked:  analysis/multiIndNormal.Rmd
    Untracked:  data/EMTAB5061/
    Untracked:  data/GSE107585_Mouse_kidney_single_cell_datamatrix.txt.gz
    Untracked:  data/GSE129788/
    Untracked:  data/GSE141115/
    Untracked:  data/GSE81547/
    Untracked:  data/GSE81608/
    Untracked:  data/GSE85241/
    Untracked:  data/MuSiC/
    Untracked:  data/baron-human.rds
    Untracked:  data/cell_type_annotation_Cels2016.csv
    Untracked:  data/cibersort/
    Untracked:  data/muraro.rds
    Untracked:  data/pbmc.txt
    Untracked:  data/pbmc_mat.RData
    Untracked:  data/pbmc_mat.rds
    Untracked:  data/robots.txt
    Untracked:  output/Xin_bycelltype_hc2.RData
    Untracked:  output/Xin_bycelltype_hc3.RData
    Untracked:  output/Xin_sampleM_hc2.RData
    Untracked:  output/Xin_sampleM_hc3.RData
    Untracked:  output/Xin_sampleM_unadj.RData
    Untracked:  output/Xin_universal_hc0.RData
    Untracked:  output/Xin_universal_hc2.RData
    Untracked:  output/Xin_universal_hc3.RData
    Untracked:  output/bulkref_pbmc_allgene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls5_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls5_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls05_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls10_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls50_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls10_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls05_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_addw_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_addw_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/results_nsclc.RData
    Untracked:  output/scref_multiInd100_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_multiInd100_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown_Nosigma2.RData
    Untracked:  output/scref_multiInd10_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_multiInd10_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown_Nosigma2.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/twoBulk_bulkref_simu_G1000_K4_refls30.RData
    Untracked:  output/twoBulk_bulkref_simu_G500_K4_refls30.RData
    Untracked:  output/twoBulk_multiscref_simu_G1000_K4.RData
    Untracked:  output/twoBulk_null_bulkref_simu_G1000_K4_refls30.RData
    Untracked:  output/twoBulk_null_multiscref_simu_G1000_K4.RData
    Untracked:  output/twoBulk_null_scref_simu_G1000_K4.RData
    Untracked:  output/twoBulk_scref_simu_G1000_K4.RData

Unstaged changes:
    Modified:   analysis/real_data_simu.Rmd
    Modified:   code/deconference_setdata.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/cellsize.Rmd</code>) and HTML (<code>docs/cellsize.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/ea758b6b29abf5690e9a2a4e465bd69915b150c4/analysis/cellsize.Rmd" target="_blank">ea758b6</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-08-20
</td>
<td>
wflow_publish(“analysis/cellsize.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Explore the difference of cell sizes of UMI/non-UMI methods.</p>
<p>Xin: rpkm and raw counts available</p>
<p>segerstope: rpkm and raw counts available</p>
<p>muraro: Transcript counts were then adjusted to the expected number of molecules based on counts, 256 possible UMI’s and poissonian counting statistics.</p>
<p>baron: raw counts</p>
<pre class="r"><code>library(MuSiC)</code></pre>
<pre><code>Loading required package: nnls</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code>library(xbioc)</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind,
    colMeans, colnames, colSums, dirname, do.call, duplicated,
    eval, evalq, Filter, Find, get, grep, grepl, intersect,
    is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
    paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
    Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which, which.max,
    which.min</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre class="r"><code>library(MASS)</code></pre>
<pre><code>
Attaching package: &#39;MASS&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:AnnotationDbi&#39;:

    select</code></pre>
<pre class="r"><code>source(&#39;code/deconference_main.R&#39;)
XinT2Deset &lt;- readRDS(&quot;data/MuSiC/XinT2Deset.rds&quot;)
xin &lt;- readRDS(&quot;data/GSE81608/xin.rds&quot;)
segerstolpe &lt;- readRDS(&quot;data/MuSiC/segerstolpe.rds&quot;)
muraro &lt;- readRDS(&quot;data/muraro.rds&quot;)
baron &lt;- readRDS(&quot;~/deconference/data/baron-human.rds&quot;)
GSE50244bulkeset &lt;- readRDS(&quot;data/MuSiC/GSE50244bulkeset.rds&quot;)
# common genes
common_genes = intersect(intersect(rownames(exprs(XinT2Deset)), rownames(segerstolpe)), intersect(unlist(lapply(strsplit(rownames(muraro),split=&#39;__&#39;),function(z){z[1]})), rownames(baron)))</code></pre>
<pre><code>Loading required package: SingleCellExperiment</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: DelayedArray</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>Loading required package: BiocParallel</code></pre>
<pre><code>
Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    aperm, apply</code></pre>
<pre class="r"><code>common_genes = (intersect(rownames(GSE50244bulkeset),common_genes))
common_cells = c(&#39;alpha&#39;,&#39;beta&#39;,&#39;delta&#39;,&#39;gamma&#39;)

length(common_genes)</code></pre>
<pre><code>[1] 16066</code></pre>
<p>baron data set</p>
<pre class="r"><code>baron_gene = match(common_genes,rownames(baron))
baron_cell = which(baron$cell_type1%in%common_cells)
out_baron = scRef_multi_proc(counts(baron)[baron_gene,baron_cell],baron$cell_type1[baron_cell],baron$human[baron_cell],estimator=&#39;separate&#39;,eps=0,
                            est_sigma2=F,sigma2=NULL,tau2=NULL,meta_var=&#39;plug_in&#39;,meta_mode = &quot;universal&quot;,cell_types = common_cells)

out_baron$S_glm</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.1487691 1.1134982 0.8432499 </code></pre>
<pre class="r"><code>out_baron$S</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.1460726 1.1164563 0.8372908 </code></pre>
<p>muraro data set</p>
<pre class="r"><code>muraro_gene =  match(common_genes,unlist(lapply(strsplit(rownames(muraro),split=&#39;__&#39;),function(z){z[1]})))
muraro_cell = which(muraro$cell_type1%in%common_cells)
out_muraro = scRef_multi_proc(normcounts(muraro)[muraro_gene,muraro_cell],muraro$cell_type1[muraro_cell],muraro$donor[muraro_cell],estimator=&#39;separate&#39;,eps=0,
                            est_sigma2=F,sigma2=NULL,tau2=NULL,meta_var=&#39;plug_in&#39;,meta_mode = &quot;universal&quot;,cell_types = common_cells)

out_muraro$S_glm</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.1697045 0.8507336 0.6296319 </code></pre>
<pre class="r"><code>out_muraro$S</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.2189782 0.8394014 0.6115425 </code></pre>
<p>Xin data set</p>
<p>Use raw counts:</p>
<pre class="r"><code>xin_gene =  match(common_genes,rownames(exprs(XinT2Deset)))
xin_cell = which(XinT2Deset$cellType%in%common_cells)
out_xin = scRef_multi_proc(exprs(XinT2Deset)[xin_gene,xin_cell],XinT2Deset$cellType[xin_cell],XinT2Deset$SubjectName[xin_cell],estimator=&#39;separate&#39;,eps=0,
                            est_sigma2=F,sigma2=NULL,tau2=NULL,meta_var=&#39;plug_in&#39;,meta_mode = &quot;universal&quot;,cell_types = common_cells)

out_xin$S_glm</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.0346887 1.0248433 0.8527105 </code></pre>
<pre class="r"><code>out_xin$S</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.0247103 1.0272805 0.8662465 </code></pre>
<p>Use raw counts adjusted by gene length:</p>
<pre class="r"><code>xin_gene2 = match(common_genes,rownames(normcounts(xin)))
xin_cell2 = which(xin$cell_type1%in%common_cells)
out_xin2 = scRef_multi_proc(t(t(normcounts(xin)[xin_gene,xin_cell]*1e-9)*c(colSums(exprs(XinT2Deset)[xin_gene,xin_cell]))),
                            xin$cell_type1[xin_cell2],xin$donor.id[xin_cell2],estimator=&#39;separate&#39;,eps=0,
                            est_sigma2=F,sigma2=NULL,tau2=NULL,meta_var=&#39;plug_in&#39;,meta_mode = &quot;universal&quot;,cell_types = common_cells)

out_xin2$S_glm</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 0.8052437 1.0477142 0.8696516 </code></pre>
<pre class="r"><code>out_xin2$S</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 0.8023332 1.0652933 0.8797034 </code></pre>
<p>segerstolpe data raw counts:</p>
<pre class="r"><code>segerstolpe_gene =  match(common_genes,unlist(lapply(strsplit(rownames(segerstolpe),split=&#39;__&#39;),function(z){z[1]})))
segerstolpe_cell = which(segerstolpe$cell_type1%in%common_cells)
out_segerstolpe = scRef_multi_proc(counts(segerstolpe)[segerstolpe_gene,segerstolpe_cell],
                                   segerstolpe$cell_type1[segerstolpe_cell],segerstolpe$age[segerstolpe_cell],estimator=&#39;separate&#39;,eps=0,
                                   est_sigma2=F,sigma2=NULL,tau2=NULL,meta_var=&#39;plug_in&#39;,meta_mode = &quot;universal&quot;,cell_types = common_cells)

out_segerstolpe$S_glm</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 0.7603227 0.9329263 0.7762657 </code></pre>
<pre class="r"><code>out_segerstolpe$S</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 0.7715393 0.9115662 0.7152070 </code></pre>
</div>
<div id="combine" class="section level2">
<h2>combine</h2>
<p>combine baron, muraro, xin2</p>
<pre class="r"><code>S_mat = rbind(out_baron$S_mat,out_muraro$S_mat,out_xin2$S_mat)

S = colMeans(S_mat,na.rm = TRUE)
S = S/S[1]

  # ## method 2: glm
  #
  S_mat_dataframe = data.frame(y = c(S_mat),
                               indi = as.factor(rep(1:nrow(S_mat),ncol(S_mat))),
                               type = as.factor(rep(common_cells,each = nrow(S_mat))))
  fit = glm.nb(y~.,S_mat_dataframe)
  S_glm = exp(fit$coefficients[-c(1:nrow(S_mat))])
  S_glm = c(1,S_glm)
  names(S_glm) = common_cells
  
  S</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 1.1807739 0.9725510 0.7192788 </code></pre>
<pre class="r"><code>  S_glm</code></pre>
<pre><code>    alpha      beta     delta     gamma 
1.0000000 0.9148313 1.0206902 0.8193664 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] SingleCellExperiment_1.4.0  SummarizedExperiment_1.12.0
 [3] DelayedArray_0.8.0          BiocParallel_1.16.0        
 [5] matrixStats_0.54.0          GenomicRanges_1.34.0       
 [7] GenomeInfoDb_1.18.1         MASS_7.3-51.1              
 [9] xbioc_0.1.18                AnnotationDbi_1.44.0       
[11] IRanges_2.16.0              S4Vectors_0.20.1           
[13] Biobase_2.42.0              BiocGenerics_0.28.0        
[15] MuSiC_0.1.1                 ggplot2_3.1.1              
[17] nnls_1.4                   

loaded via a namespace (and not attached):
 [1] bit64_0.9-7            assertthat_0.2.0       BiocManager_1.30.10   
 [4] highr_0.7              blob_1.1.1             GenomeInfoDbData_1.2.0
 [7] yaml_2.2.0             pillar_1.3.1           RSQLite_2.1.1         
[10] backports_1.1.2        lattice_0.20-38        quantreg_5.36         
[13] glue_1.3.0             digest_0.6.18          promises_1.0.1        
[16] XVector_0.22.0         colorspace_1.3-2       htmltools_0.3.6       
[19] httpuv_1.4.5           Matrix_1.2-15          plyr_1.8.4            
[22] pkgconfig_2.0.2        bibtex_0.4.2           SparseM_1.78          
[25] zlibbioc_1.28.0        purrr_0.3.2            xtable_1.8-3          
[28] scales_1.0.0           whisker_0.3-2          later_0.7.5           
[31] MatrixModels_0.4-1     git2r_0.26.1           tibble_2.1.1          
[34] pkgmaker_0.27          withr_2.1.2            lazyeval_0.2.1        
[37] magrittr_1.5           crayon_1.3.4           memoise_1.1.0         
[40] mcmc_0.9-7             evaluate_0.12          fs_1.3.1              
[43] tools_3.5.1            registry_0.5           stringr_1.3.1         
[46] MCMCpack_1.4-4         munsell_0.5.0          compiler_3.5.1        
[49] rlang_0.4.0            grid_3.5.1             RCurl_1.95-4.11       
[52] bitops_1.0-6           rmarkdown_1.10         gtable_0.2.0          
[55] codetools_0.2-15       DBI_1.0.0              reshape2_1.4.3        
[58] R6_2.3.0               knitr_1.20             dplyr_0.8.0.1         
[61] bit_1.1-14             workflowr_1.6.2        rprojroot_1.3-2       
[64] stringi_1.2.4          Rcpp_1.0.4.6           tidyselect_0.2.5      
[67] coda_0.19-2           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
