<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2021-04-09" />

<title>Simulation on real data set, with comparisons to MuSiC and CibersortX</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">deconference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="about.html">Data</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/DongyueXie/deconference">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Simulation on real data set, with comparisons to MuSiC and CibersortX</h1>
<h4 class="author">Dongyue Xie</h4>
<h4 class="date">2021-04-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>deconference/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200403code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200403)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200403code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200403)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetree2282a21dba9aa633ebd61f135158fc38f44a7e42targetblank2282a21a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DongyueXie/deconference/tree/2282a21dba9aa633ebd61f135158fc38f44a7e42" target="_blank">2282a21</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetree2282a21dba9aa633ebd61f135158fc38f44a7e42targetblank2282a21a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DongyueXie/deconference/tree/2282a21dba9aa633ebd61f135158fc38f44a7e42" target="_blank">2282a21</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  code/prepare_data_for_cibersort.R
    Untracked:  code/wols.R
    Untracked:  data/MuSiC/
    Untracked:  data/pancreas/

Unstaged changes:
    Modified:   analysis/MuSiC.Rmd
    Modified:   analysis/gene_weights.Rmd
    Modified:   code/deconference_estfunc.R
    Modified:   code/deconference_meta.R
    Modified:   code/deconference_setdata.R
    Modified:   code/utils.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gene_weights_addmusic.Rmd</code>) and HTML (<code>docs/gene_weights_addmusic.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/2282a21dba9aa633ebd61f135158fc38f44a7e42/analysis/gene_weights_addmusic.Rmd" target="_blank">2282a21</a>
</td>
<td>
Dongyue Xie
</td>
<td>
2021-04-22
</td>
<td>
wflow_publish(“analysis/gene_weights_addmusic.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DongyueXie/deconference/d8a41e2d2fd333f32a9b231bee2fd3ce3c685b0c/docs/gene_weights_addmusic.html" target="_blank">d8a41e2</a>
</td>
<td>
Dongyue Xie
</td>
<td>
2021-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/fe2b130e9f1acb3137b26f88035ed1c433942f50/analysis/gene_weights_addmusic.Rmd" target="_blank">fe2b130</a>
</td>
<td>
Dongyue Xie
</td>
<td>
2021-04-22
</td>
<td>
wflow_publish(“analysis/gene_weights_addmusic.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>See <a href="gene_weights.html">here</a> for a previous simulation on choosing gene weights.</p>
<p>In MuSiC:</p>
<p>The default is for bulk data Y: convert y such that it sums to 1 and then multiply y by 100; for reference matrix X: it’s the relative expression matrix whose each column sum to 1; it’s default mode also multiply X by cell size S, where S is very big if estimated from single cell data.</p>
<p>The above scaling makes the parameter nu in the weight computation important, where <span class="math inline">\(weight = 1/(\nu+r^2+across.indi.var)\)</span>.</p>
<pre class="r"><code>library(MuSiC)</code></pre>
<pre><code>Loading required package: nnls</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code>library(Biobase)</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre class="r"><code>library(xbioc)</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre class="r"><code>source(&#39;code/deconference_main.R&#39;)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    rowMedians</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>source(&#39;code/utils.R&#39;)
source(&#39;code/wols.R&#39;)
seger &lt;- readRDS(&quot;data/pancreas/segerstolpe_raw.rds&quot;)

#XinT2D.eset &lt;- readRDS(&quot;data/MuSiC/XinT2Deset.rds&quot;)
#XinT2D.construct.full = bulk_construct(XinT2D.eset, clusters = &#39;cellType&#39;, samples = &#39;SubjectName&#39;)
#XinT2D.construct.full$prop.real = relative.ab(XinT2D.construct.full$num.real, by.col = FALSE)
xin_raw &lt;- readRDS(&quot;data/pancreas/xin_raw.rds&quot;)
baron = readRDS(&quot;data/pancreas/baron.rds&quot;)</code></pre>
<pre class="r"><code>bulk_mat_to_es = function(Y){
  ExpressionSet(assayData = Y)
}


#&#39;@param b either a vector or a matrix of dimension K*n_bulk
createBulk_no.tech.noise = function(X_array, b,gene_names){
  if(is.null(dim(b))){
    y = apply(X_array,3,function(z){z%*%b})
  }else{
    G = dim(X_array)[1]
    K = dim(X_array)[2]
    Ni = dim(X_array)[3]
    y = matrix(nrow=G,ncol=Ni)
    for(i in 1:Ni){
      y[,i] = X_array[,,i]%*%b[,i]
    }
  }
  rownames(y) = gene_names
  colnames(y) = 1:ncol(y)
  list(y=y,b=b)
}

createBulk_tech.noise = function(X_array, b, gene_names, ls_per_gene = 500,seed=12345){
  
  G = dim(X_array)[1]
  K = dim(X_array)[2]
  Ni = dim(X_array)[3]
    
  if(is.null(dim(b))){
    y = apply(X_array,3,function(z){z%*%b})
  }else{
    y = matrix(nrow=G,ncol=Ni)
    for(i in 1:Ni){
      y[,i] = X_array[,,i]%*%b[,i]
    }
  }
  y = y/colSums(y)*ls_per_gene*G
  set.seed(seed)
  y = matrix(rpois(prod(dim(y)),y),nrow=nrow(y),ncol=ncol(y))
  rownames(y) = gene_names
  colnames(y) = 1:ncol(y)
  list(y=y,b=b*ls_per_gene)
}


calc_weight = function(){
  ####
}





is.covered = function(b,bhat,bhat.se,alpha=0.05){
  (b&lt;=bhat+qnorm(1-alpha/2)*bhat.se)&amp;(b&gt;=bhat-qnorm(1-alpha/2)*bhat.se)
}

#&#39;@param design.mat output from scRef_multi_proc, in which X is the reference matrix.
#&#39;@param datax output from set_data_decon, in which y is the bulk data matrix.
#&#39;@param b betas, K by n_bulk.


diag.cov = function(X){
  #center X
  X = scale(X,center=TRUE,scale=F)
  colSums(X^2)/(nrow(X)-1)
}</code></pre>
</div>
<div id="create-bulk-datawith-technical-noise-using-xin-dataset." class="section level2">
<h2>Create bulk data(with technical noise) using Xin dataset.</h2>
<pre class="r"><code>cell_types = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;)
K = length(cell_types)
rm.indi = c(&quot;Non T2D 4&quot;,&quot;Non T2D 7&quot;,&quot;Non T2D 10&quot;,&quot;Non T2D 12&quot;)
#rm.indi = levels(XinT2D.eset$SubjectName)[rm.indi]
rm.indi.idx = which(xin_raw$individual%in%rm.indi)
datax.xin = set_data_decon(Y = xin_raw[,-rm.indi.idx],cell_types = cell_types, gene_thresh = 0,max_count_quantile = 1,w=1)
design.mat.xin = scRef_multi_proc(datax.xin$Y,datax.xin$cell_type_idx,datax.xin$indi_idx,estimator=&quot;separate&quot;,est_sigma2 = FALSE)
# create bulk data
b = c(0.1,0.1,0.3,0.5)
bulk.xin = createBulk_tech.noise(design.mat.xin$X_array,b,gene_names = rownames(xin_raw),seed=12345,ls_per_gene = 500)</code></pre>
<div id="seger-as-reference" class="section level3">
<h3>Seger as reference</h3>
<pre class="r"><code># fit music
seger.es = sce_to_es(seger)
cell_size = data.frame(type = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;), size=c(1,1,1,1))
fit.music = music_prop(bulk_mat_to_es(bulk.xin$y),seger.es,clusters = &#39;cell_type&#39;, samples = &#39;individual&#39;,
                       select.ct = cell_types,cell_size=cell_size,verbose = FALSE,nu=1e-4,normalize = F)
rmse(fit.music$Est.prop.weighted,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1568751</code></pre>
<pre class="r"><code>round(fit.music$Est.prop.weighted,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.05 0.21  0.10  0.64
2   0.03 0.17  0.10  0.70
3   0.11 0.03  0.19  0.67
4   0.10 0.05  0.25  0.60
5   0.19 0.07  0.19  0.56
6   0.09 0.02  0.30  0.59
7   0.09 0.01  0.15  0.76
8   0.07 0.02  0.17  0.74
9   0.05 0.02  0.16  0.77
10  0.08 0.02  0.14  0.76
11  0.14 0.00  0.16  0.70
12  0.13 0.08  0.14  0.65
13  0.62 0.02  0.32  0.04
14  0.14 0.00  0.10  0.75</code></pre>
<pre class="r"><code>fit.music2 = music_prop(bulk_mat_to_es(bulk.xin$y),seger.es,clusters = &#39;cell_type&#39;, samples = &#39;individual&#39;,
                       select.ct = cell_types,cell_size=cell_size,verbose = FALSE,nu=1e-4,normalize = T)
rmse(fit.music2$Est.prop.weighted,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1369467</code></pre>
<pre class="r"><code>round(fit.music2$Est.prop.weighted,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.10 0.31  0.26  0.33
2   0.09 0.29  0.23  0.39
3   0.21 0.13  0.21  0.46
4   0.19 0.07  0.29  0.45
5   0.47 0.12  0.27  0.14
6   0.13 0.15  0.34  0.38
7   0.16 0.18  0.16  0.49
8   0.27 0.16  0.19  0.39
9   0.09 0.16  0.26  0.49
10  0.12 0.14  0.34  0.40
11  0.21 0.09  0.20  0.49
12  0.33 0.19  0.20  0.27
13  0.38 0.22  0.36  0.05
14  0.21 0.08  0.13  0.58</code></pre>
<pre class="r"><code>fit.music3 = music_prop(bulk_mat_to_es(bulk.xin$y),seger.es,clusters = &#39;cell_type&#39;, samples = &#39;individual&#39;,
                       select.ct = cell_types,cell_size=cell_size,verbose = FALSE,nu=1e-2,normalize = T)
rmse(fit.music3$Est.prop.weighted,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1112698</code></pre>
<pre class="r"><code>round(fit.music3$Est.prop.weighted,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.09 0.30  0.10  0.51
2   0.10 0.33  0.09  0.48
3   0.13 0.12  0.18  0.58
4   0.11 0.13  0.23  0.53
5   0.24 0.09  0.16  0.51
6   0.13 0.10  0.27  0.51
7   0.10 0.24  0.14  0.52
8   0.27 0.15  0.17  0.41
9   0.07 0.18  0.15  0.59
10  0.08 0.17  0.12  0.63
11  0.16 0.08  0.17  0.59
12  0.20 0.17  0.14  0.48
13  0.37 0.12  0.27  0.24
14  0.22 0.13  0.11  0.54</code></pre>
<pre class="r"><code>datax = set_data_decon(y = bulk.xin$y, Y = seger,
                       cell_types = cell_types, 
                       gene_thresh = 0,max_count_quantile = 1,w=1)

design.mat = scRef_multi_proc(datax$Y,datax$cell_type_idx,datax$indi_idx,
                                    estimator=&quot;separate&quot;,est_sigma2 = FALSE)

fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
rmse(fit.my$beta_hat,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1184853</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.17 0.23 0.20 0.40
 [2,] 0.16 0.22 0.25 0.38
 [3,] 0.11 0.12 0.26 0.51
 [4,] 0.13 0.08 0.39 0.40
 [5,] 0.33 0.13 0.38 0.16
 [6,] 0.12 0.15 0.42 0.31
 [7,] 0.16 0.16 0.18 0.50
 [8,] 0.16 0.21 0.18 0.45
 [9,] 0.10 0.14 0.24 0.51
[10,] 0.16 0.16 0.30 0.38
[11,] 0.16 0.12 0.15 0.57
[12,] 0.24 0.15 0.29 0.32
[13,] 0.36 0.15 0.42 0.07
[14,] 0.19 0.09 0.17 0.55</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.011 0.011 0.016 0.014
 [2,] 0.014 0.013 0.023 0.019
 [3,] 0.011 0.009 0.020 0.017
 [4,] 0.009 0.006 0.018 0.017
 [5,] 0.014 0.010 0.019 0.015
 [6,] 0.011 0.009 0.018 0.015
 [7,] 0.014 0.011 0.020 0.019
 [8,] 0.015 0.013 0.020 0.019
 [9,] 0.010 0.008 0.017 0.015
[10,] 0.011 0.009 0.018 0.015
[11,] 0.012 0.009 0.019 0.018
[12,] 0.011 0.008 0.017 0.015
[13,] 0.016 0.010 0.021 0.016
[14,] 0.011 0.007 0.014 0.014</code></pre>
<pre class="r"><code>is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE FALSE
 [2,] FALSE FALSE FALSE FALSE
 [3,]  TRUE  TRUE FALSE  TRUE
 [4,] FALSE FALSE FALSE FALSE
 [5,] FALSE FALSE FALSE FALSE
 [6,]  TRUE FALSE FALSE FALSE
 [7,] FALSE FALSE FALSE  TRUE
 [8,] FALSE FALSE FALSE FALSE
 [9,]  TRUE FALSE FALSE  TRUE
[10,] FALSE FALSE  TRUE FALSE
[11,] FALSE FALSE FALSE FALSE
[12,] FALSE FALSE  TRUE FALSE
[13,] FALSE FALSE FALSE FALSE
[14,] FALSE  TRUE FALSE FALSE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.1785714</code></pre>
<pre class="r"><code>diag(cor(fit.my$weight,fit.music$Weight.gene,use = &quot;complete.obs&quot;))</code></pre>
<pre><code> [1] 0.1059406 0.1457204 0.2318830 0.1805058 0.1846132 0.1075222 0.2037898
 [8] 0.1360044 0.1067047 0.1050186 0.1498726 0.2279809 0.1448036 0.1569034</code></pre>
<pre class="r"><code>diag(cor(fit.my$X,fit.music$X,use = &quot;pairwise.complete.obs&quot;))</code></pre>
<pre><code>    alpha      beta     delta     gamma 
0.9998892 0.9999562 0.9999928 0.9996858 </code></pre>
<pre class="r"><code>diag(cor(fit.my$Vg[,c(1,6,11,16)],fit.music$Sigma,use = &quot;pairwise.complete.obs&quot;))</code></pre>
<pre><code>[1] 0.9998389 0.9999864 0.9999957 0.9996672</code></pre>
<p>scale y as y/sum(y)*100</p>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = T,use.weight.for.var = T)
rmse(fit.my$beta_hat,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.09512512</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.05 0.30 0.19 0.46
 [2,] 0.08 0.26 0.19 0.47
 [3,] 0.07 0.11 0.26 0.55
 [4,] 0.09 0.08 0.39 0.43
 [5,] 0.26 0.15 0.34 0.26
 [6,] 0.08 0.12 0.39 0.40
 [7,] 0.09 0.17 0.20 0.54
 [8,] 0.08 0.21 0.21 0.50
 [9,] 0.02 0.16 0.26 0.56
[10,] 0.11 0.16 0.27 0.46
[11,] 0.09 0.08 0.19 0.64
[12,] 0.15 0.18 0.22 0.45
[13,] 0.28 0.13 0.39 0.20
[14,] 0.14 0.07 0.15 0.63</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.037 0.040 0.055 0.052
 [2,] 0.038 0.048 0.058 0.062
 [3,] 0.026 0.025 0.042 0.037
 [4,] 0.016 0.020 0.060 0.049
 [5,] 0.044 0.034 0.072 0.063
 [6,] 0.032 0.022 0.039 0.039
 [7,] 0.030 0.030 0.036 0.037
 [8,] 0.038 0.026 0.033 0.035
 [9,] 0.031 0.028 0.048 0.045
[10,] 0.033 0.036 0.071 0.063
[11,] 0.035 0.021 0.020 0.039
[12,] 0.027 0.035 0.045 0.047
[13,] 0.040 0.030 0.056 0.056
[14,] 0.024 0.021 0.031 0.034</code></pre>
<pre class="r"><code>is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,]  TRUE FALSE FALSE  TRUE
 [2,]  TRUE FALSE  TRUE  TRUE
 [3,]  TRUE  TRUE  TRUE  TRUE
 [4,]  TRUE  TRUE  TRUE  TRUE
 [5,] FALSE  TRUE  TRUE FALSE
 [6,]  TRUE  TRUE FALSE FALSE
 [7,]  TRUE FALSE FALSE  TRUE
 [8,]  TRUE FALSE FALSE  TRUE
 [9,] FALSE FALSE  TRUE  TRUE
[10,]  TRUE  TRUE  TRUE  TRUE
[11,]  TRUE  TRUE FALSE FALSE
[12,]  TRUE FALSE  TRUE  TRUE
[13,] FALSE  TRUE  TRUE FALSE
[14,]  TRUE  TRUE FALSE FALSE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.6428571</code></pre>
<pre class="r"><code>diag(cor(fit.my$weight,fit.music$Weight.gene,use = &quot;complete.obs&quot;))</code></pre>
<pre><code> [1] 0.3320024 0.3751053 0.3749012 0.3690007 0.3634617 0.3817499 0.3981354
 [8] 0.3636165 0.3720600 0.3468162 0.3914478 0.3642222 0.4019095 0.3554792</code></pre>
<p>Scaling of y, selection of nu is tricky.</p>
<p>Use music X,weights,variance to calculate weights:</p>
<pre class="r"><code># use everything from music
# music weights
fit.my = wols(fit.music$X,datax$y,fit.music$Sigma,b=t(as.matrix(fit.music$Est.prop.weighted))*100,
              w.mode = &#39;res+bvb&#39;,adj.v = F,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),nu=1e-4,scale.y = T)
rmse(fit.my$beta_hat,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1211295</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.03 0.29 0.14 0.54
 [2,] 0.03 0.24 0.14 0.59
 [3,] 0.09 0.05 0.30 0.57
 [4,] 0.08 0.07 0.37 0.48
 [5,] 0.15 0.12 0.28 0.45
 [6,] 0.07 0.02 0.38 0.53
 [7,] 0.07 0.01 0.24 0.68
 [8,] 0.06 0.04 0.29 0.61
 [9,] 0.05 0.03 0.24 0.68
[10,] 0.06 0.04 0.23 0.67
[11,] 0.11 0.00 0.22 0.67
[12,] 0.09 0.15 0.17 0.59
[13,] 0.49 0.03 0.42 0.06
[14,] 0.13 0.01 0.16 0.70</code></pre>
<pre class="r"><code>diag(cor(fit.my$weight,fit.music$Weight.gene,use = &quot;complete.obs&quot;))</code></pre>
<pre><code> [1] 0.3652590 0.4127070 0.3879344 0.3853449 0.3818779 0.3930913 0.4189878
 [8] 0.3874375 0.3958587 0.3792553 0.4135212 0.3898178 0.4344387 0.4104707</code></pre>
<p>other hc2, hc3</p>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc2&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE FALSE
 [2,] FALSE FALSE FALSE FALSE
 [3,]  TRUE  TRUE FALSE  TRUE
 [4,] FALSE FALSE FALSE FALSE
 [5,] FALSE FALSE FALSE FALSE
 [6,]  TRUE FALSE FALSE FALSE
 [7,] FALSE FALSE FALSE  TRUE
 [8,] FALSE FALSE FALSE FALSE
 [9,]  TRUE FALSE FALSE  TRUE
[10,] FALSE FALSE  TRUE FALSE
[11,] FALSE FALSE FALSE FALSE
[12,] FALSE FALSE  TRUE FALSE
[13,] FALSE FALSE FALSE FALSE
[14,] FALSE  TRUE FALSE FALSE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.1785714</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.011 0.011 0.016 0.014
 [2,] 0.014 0.013 0.023 0.019
 [3,] 0.011 0.009 0.020 0.017
 [4,] 0.009 0.006 0.018 0.017
 [5,] 0.014 0.010 0.019 0.015
 [6,] 0.011 0.009 0.018 0.015
 [7,] 0.014 0.011 0.020 0.019
 [8,] 0.015 0.013 0.020 0.019
 [9,] 0.011 0.008 0.017 0.015
[10,] 0.011 0.009 0.018 0.015
[11,] 0.012 0.009 0.019 0.018
[12,] 0.011 0.008 0.017 0.015
[13,] 0.016 0.010 0.021 0.016
[14,] 0.011 0.007 0.014 0.014</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc3&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE FALSE
 [2,] FALSE FALSE FALSE FALSE
 [3,]  TRUE  TRUE FALSE  TRUE
 [4,] FALSE FALSE FALSE FALSE
 [5,] FALSE FALSE FALSE FALSE
 [6,]  TRUE FALSE FALSE FALSE
 [7,] FALSE FALSE FALSE  TRUE
 [8,] FALSE FALSE FALSE FALSE
 [9,]  TRUE FALSE FALSE  TRUE
[10,] FALSE FALSE  TRUE FALSE
[11,] FALSE FALSE FALSE FALSE
[12,] FALSE FALSE  TRUE FALSE
[13,] FALSE FALSE FALSE FALSE
[14,] FALSE  TRUE FALSE FALSE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.1785714</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.011 0.011 0.016 0.014
 [2,] 0.015 0.014 0.024 0.019
 [3,] 0.011 0.009 0.020 0.017
 [4,] 0.009 0.006 0.018 0.017
 [5,] 0.014 0.010 0.019 0.015
 [6,] 0.011 0.009 0.018 0.015
 [7,] 0.014 0.011 0.020 0.019
 [8,] 0.015 0.013 0.020 0.019
 [9,] 0.011 0.008 0.017 0.015
[10,] 0.011 0.009 0.018 0.015
[11,] 0.012 0.009 0.019 0.018
[12,] 0.011 0.008 0.017 0.015
[13,] 0.016 0.010 0.022 0.016
[14,] 0.011 0.007 0.014 0.014</code></pre>
<p>do not use weights when calculating variance</p>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1] [,2]  [,3]  [,4]
 [1,]  TRUE TRUE  TRUE  TRUE
 [2,]  TRUE TRUE  TRUE  TRUE
 [3,]  TRUE TRUE  TRUE  TRUE
 [4,]  TRUE TRUE  TRUE  TRUE
 [5,]  TRUE TRUE  TRUE FALSE
 [6,]  TRUE TRUE  TRUE  TRUE
 [7,]  TRUE TRUE FALSE  TRUE
 [8,]  TRUE TRUE  TRUE  TRUE
 [9,]  TRUE TRUE  TRUE  TRUE
[10,]  TRUE TRUE  TRUE  TRUE
[11,]  TRUE TRUE FALSE  TRUE
[12,] FALSE TRUE  TRUE FALSE
[13,]  TRUE TRUE  TRUE FALSE
[14,] FALSE TRUE FALSE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.8571429</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.092 0.189 0.097 0.162
 [2,] 0.095 0.176 0.104 0.125
 [3,] 0.052 0.114 0.067 0.076
 [4,] 0.031 0.072 0.069 0.055
 [5,] 0.141 0.119 0.130 0.063
 [6,] 0.065 0.153 0.161 0.113
 [7,] 0.075 0.150 0.059 0.117
 [8,] 0.089 0.189 0.090 0.123
 [9,] 0.063 0.140 0.078 0.147
[10,] 0.067 0.146 0.115 0.141
[11,] 0.051 0.122 0.062 0.099
[12,] 0.043 0.123 0.086 0.052
[13,] 0.215 0.143 0.184 0.067
[14,] 0.042 0.086 0.032 0.071</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc2&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1] [,2]  [,3]  [,4]
 [1,]  TRUE TRUE  TRUE  TRUE
 [2,]  TRUE TRUE  TRUE  TRUE
 [3,]  TRUE TRUE  TRUE  TRUE
 [4,]  TRUE TRUE  TRUE  TRUE
 [5,]  TRUE TRUE  TRUE FALSE
 [6,]  TRUE TRUE  TRUE  TRUE
 [7,]  TRUE TRUE FALSE  TRUE
 [8,]  TRUE TRUE  TRUE  TRUE
 [9,]  TRUE TRUE  TRUE  TRUE
[10,]  TRUE TRUE  TRUE  TRUE
[11,]  TRUE TRUE FALSE  TRUE
[12,] FALSE TRUE  TRUE FALSE
[13,]  TRUE TRUE  TRUE FALSE
[14,] FALSE TRUE FALSE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.8571429</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.092 0.189 0.098 0.162
 [2,] 0.095 0.177 0.107 0.126
 [3,] 0.053 0.115 0.067 0.077
 [4,] 0.032 0.072 0.070 0.055
 [5,] 0.142 0.120 0.132 0.063
 [6,] 0.066 0.154 0.162 0.113
 [7,] 0.075 0.151 0.059 0.118
 [8,] 0.089 0.190 0.090 0.123
 [9,] 0.064 0.140 0.078 0.147
[10,] 0.067 0.146 0.117 0.141
[11,] 0.051 0.122 0.064 0.100
[12,] 0.044 0.124 0.088 0.053
[13,] 0.216 0.143 0.184 0.067
[14,] 0.042 0.086 0.033 0.071</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc3&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1] [,2]  [,3]  [,4]
 [1,]  TRUE TRUE  TRUE  TRUE
 [2,]  TRUE TRUE  TRUE  TRUE
 [3,]  TRUE TRUE  TRUE  TRUE
 [4,]  TRUE TRUE  TRUE  TRUE
 [5,]  TRUE TRUE  TRUE FALSE
 [6,]  TRUE TRUE  TRUE  TRUE
 [7,]  TRUE TRUE FALSE  TRUE
 [8,]  TRUE TRUE  TRUE  TRUE
 [9,]  TRUE TRUE  TRUE  TRUE
[10,]  TRUE TRUE  TRUE  TRUE
[11,]  TRUE TRUE FALSE  TRUE
[12,] FALSE TRUE  TRUE FALSE
[13,]  TRUE TRUE  TRUE FALSE
[14,] FALSE TRUE FALSE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.8571429</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.093 0.190 0.100 0.163
 [2,] 0.096 0.177 0.111 0.128
 [3,] 0.053 0.115 0.068 0.077
 [4,] 0.032 0.073 0.071 0.056
 [5,] 0.142 0.120 0.133 0.064
 [6,] 0.066 0.154 0.162 0.114
 [7,] 0.075 0.151 0.059 0.118
 [8,] 0.089 0.190 0.090 0.123
 [9,] 0.064 0.140 0.079 0.148
[10,] 0.068 0.147 0.118 0.142
[11,] 0.052 0.122 0.067 0.101
[12,] 0.044 0.124 0.089 0.053
[13,] 0.216 0.144 0.184 0.067
[14,] 0.043 0.087 0.034 0.072</code></pre>
<pre class="r"><code>plot(fit.my$weight[,3])
quantile((fit.my$weight[,3]),na.rm=TRUE)
order(fit.my$weight[,3],decreasing = T)[1:10]

# top 10 weighted samples

round(fit.my$weight[order(fit.my$weight[,3],decreasing = T)[1:10],3],7)

round(fit.my$X[order(fit.my$weight[,3],decreasing = T)[1:10],],4)</code></pre>
<p><strong>gene weights comparison with cibersort x</strong></p>
<pre class="r"><code>seger_sig_cibersort_output = read.delim(&quot;data/pancreas/seger_sig_cibersort_output.txt&quot;, quote=&quot;&quot;)
sig_genes = seger_sig_cibersort_output[,1]
length(sig_genes)</code></pre>
<pre><code>[1] 1726</code></pre>
<pre class="r"><code>genes_all = rownames(fit.music$Weight.gene)
w_cibersort = rep(0,length(genes_all))
names(w_cibersort) = genes_all
w_cibersort[names(w_cibersort)%in%sig_genes] = 1


cor(fit.my$weight[,1],w_cibersort,use=&#39;pairwise.complete.obs&#39;)</code></pre>
<pre><code>[1] -0.04965981</code></pre>
<pre class="r"><code>cor(fit.music$Weight.gene[,1],w_cibersort,use=&#39;pairwise.complete.obs&#39;)</code></pre>
<pre><code>[1] -0.1087975</code></pre>
<pre class="r"><code>cor(fit.my$weight[,2],w_cibersort,use=&#39;pairwise.complete.obs&#39;)</code></pre>
<pre><code>[1] -0.04031478</code></pre>
<pre class="r"><code>cor(fit.music$Weight.gene[,2],w_cibersort,use=&#39;pairwise.complete.obs&#39;)</code></pre>
<pre><code>[1] -0.07450899</code></pre>
<pre class="r"><code>cor(fit.my$weight[,3],w_cibersort,use=&#39;pairwise.complete.obs&#39;)</code></pre>
<pre><code>[1] -0.03909558</code></pre>
<pre class="r"><code>cor(fit.music$Weight.gene[,3],w_cibersort,use=&#39;pairwise.complete.obs&#39;)</code></pre>
<pre><code>[1] -0.08095118</code></pre>
<pre class="r"><code>#plot(w_cibersort,fit.music$Weight.gene[,1])
#cibersort result

seger_cibersort_Results = read.delim(&quot;data/pancreas/seger_cibersort_Results.txt&quot;, quote=&quot;&quot;)
cibersort_est = seger_cibersort_Results[,2:5]
cibersort_est = cibersort_est[,c(2,4,1,3)]
round(cibersort_est,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.17 0.01  0.33  0.48
2   0.13 0.01  0.34  0.52
3   0.16 0.01  0.24  0.59
4   0.14 0.01  0.33  0.52
5   0.35 0.01  0.45  0.19
6   0.16 0.00  0.53  0.30
7   0.16 0.01  0.26  0.57
8   0.10 0.01  0.26  0.63
9   0.12 0.01  0.34  0.53
10  0.24 0.01  0.38  0.37
11  0.19 0.00  0.22  0.59
12  0.28 0.02  0.28  0.42
13  0.28 0.01  0.62  0.08
14  0.19 0.01  0.14  0.67</code></pre>
<pre class="r"><code>rmse(as.matrix(cibersort_est),(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1280139</code></pre>
</div>
</div>
<div id="baron-as-reference" class="section level2">
<h2>Baron as reference</h2>
<pre class="r"><code># fit music
baron.es = sce_to_es(baron)
cell_size = data.frame(type = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;), size=c(1,1,1,1))
fit.music = music_prop(bulk_mat_to_es(bulk.xin$y),baron.es,clusters = &#39;cell_type&#39;, samples = &#39;individual&#39;,
                       select.ct = cell_types,cell_size=cell_size,verbose = FALSE,nu=1e-4)
rmse(fit.music$Est.prop.weighted,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1749954</code></pre>
<pre class="r"><code>round(fit.music$Est.prop.weighted,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.44    0  0.19  0.37
2   0.36    0  0.19  0.45
3   0.21    0  0.21  0.59
4   0.19    0  0.28  0.53
5   0.62    0  0.24  0.14
6   0.24    0  0.41  0.35
7   0.23    0  0.20  0.56
8   0.14    0  0.21  0.65
9   0.21    0  0.26  0.53
10  0.40    0  0.23  0.37
11  0.19    0  0.19  0.61
12  0.39    0  0.21  0.40
13  0.61    0  0.33  0.06
14  0.25    0  0.11  0.64</code></pre>
<pre class="r"><code>fit.music2 = music_prop(bulk_mat_to_es(bulk.xin$y),baron.es,clusters = &#39;cell_type&#39;, samples = &#39;individual&#39;,
                       select.ct = cell_types,cell_size=cell_size,verbose = FALSE,nu=1e-4,normalize = TRUE)
rmse(fit.music2$Est.prop.weighted,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1623082</code></pre>
<pre class="r"><code>round(fit.music2$Est.prop.weighted,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.33 0.06  0.21  0.40
2   0.32 0.07  0.21  0.41
3   0.24 0.00  0.19  0.57
4   0.22 0.00  0.27  0.51
5   0.51 0.05  0.26  0.17
6   0.22 0.00  0.40  0.38
7   0.29 0.00  0.19  0.52
8   0.17 0.00  0.21  0.62
9   0.22 0.01  0.26  0.51
10  0.32 0.00  0.23  0.45
11  0.23 0.00  0.18  0.59
12  0.40 0.03  0.20  0.37
13  0.63 0.00  0.31  0.06
14  0.27 0.00  0.11  0.62</code></pre>
<pre class="r"><code>datax = set_data_decon(y = bulk.xin$y, Y = baron,
                       cell_types = cell_types, 
                       gene_thresh = 0,max_count_quantile = 1,w=1)

design.mat = scRef_multi_proc(datax$Y,datax$cell_type_idx,datax$indi_idx,
                                    estimator=&quot;separate&quot;,est_sigma2 = FALSE)

fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
rmse(fit.my$beta_hat,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1533415</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.30 0.02 0.21 0.46
 [2,] 0.26 0.03 0.21 0.50
 [3,] 0.17 0.01 0.21 0.61
 [4,] 0.17 0.01 0.27 0.55
 [5,] 0.56 0.00 0.28 0.15
 [6,] 0.20 0.01 0.43 0.36
 [7,] 0.21 0.01 0.19 0.59
 [8,] 0.15 0.01 0.20 0.64
 [9,] 0.18 0.02 0.26 0.55
[10,] 0.32 0.01 0.26 0.42
[11,] 0.20 0.01 0.17 0.61
[12,] 0.34 0.02 0.22 0.42
[13,] 0.56 0.01 0.36 0.07
[14,] 0.27 0.01 0.12 0.59</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.011 0.005 0.008 0.012
 [2,] 0.016 0.007 0.009 0.018
 [3,] 0.013 0.005 0.008 0.014
 [4,] 0.010 0.004 0.009 0.012
 [5,] 0.016 0.004 0.015 0.019
 [6,] 0.013 0.006 0.015 0.017
 [7,] 0.015 0.004 0.007 0.015
 [8,] 0.014 0.001 0.011 0.015
 [9,] 0.012 0.008 0.010 0.013
[10,] 0.014 0.006 0.010 0.016
[11,] 0.015 0.004 0.012 0.015
[12,] 0.014 0.005 0.009 0.015
[13,] 0.017 0.007 0.016 0.017
[14,] 0.013 0.004 0.007 0.013</code></pre>
<pre class="r"><code>is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE FALSE
 [2,] FALSE FALSE FALSE  TRUE
 [3,] FALSE FALSE FALSE FALSE
 [4,] FALSE FALSE FALSE FALSE
 [5,] FALSE FALSE  TRUE FALSE
 [6,] FALSE FALSE FALSE FALSE
 [7,] FALSE FALSE FALSE FALSE
 [8,] FALSE FALSE FALSE FALSE
 [9,] FALSE FALSE FALSE FALSE
[10,] FALSE FALSE FALSE FALSE
[11,] FALSE FALSE FALSE FALSE
[12,] FALSE FALSE FALSE FALSE
[13,] FALSE FALSE FALSE FALSE
[14,] FALSE FALSE FALSE FALSE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.03571429</code></pre>
<pre class="r"><code>diag(cor(fit.my$weight,fit.music$Weight.gene,use = &quot;complete.obs&quot;))</code></pre>
<pre><code> [1] 0.23126940 0.15345270 0.09804961 0.10234891 0.16688528 0.23576872
 [7] 0.15695752 0.09831418 0.12116898 0.23082402 0.08268900 0.29589836
[13] 0.11608786 0.15464393</code></pre>
<pre class="r"><code>diag(cor(fit.my$X,fit.music$X,use = &quot;complete.obs&quot;))</code></pre>
<pre><code>    alpha      beta     delta     gamma 
0.9998916 0.9999801 0.9999975 0.9999882 </code></pre>
<pre class="r"><code>diag(cor(fit.my$Vg[,c(1,6,11,16)],fit.music$Sigma,use = &quot;complete.obs&quot;))</code></pre>
<pre><code>[1] 0.9998713 0.9999977 0.9997575 0.9999945</code></pre>
<pre class="r"><code># use everything from music
# music weights
fit.my = wols(fit.music$X,datax$y,fit.music$Sigma,b=t(as.matrix(fit.music$Est.prop.weighted))*100,
              w.mode = &#39;res+bvb&#39;,adj.v = F,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),nu=1e-4,scale.y = T)
rmse(fit.my$beta_hat,(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1418511</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.32    0 0.33 0.35
 [2,] 0.25    0 0.33 0.42
 [3,] 0.17    0 0.36 0.48
 [4,] 0.18    0 0.35 0.47
 [5,] 0.47    0 0.41 0.11
 [6,] 0.15    0 0.50 0.35
 [7,] 0.19    0 0.30 0.51
 [8,] 0.12    0 0.38 0.50
 [9,] 0.17    0 0.32 0.51
[10,] 0.26    0 0.34 0.40
[11,] 0.18    0 0.29 0.54
[12,] 0.30    0 0.32 0.38
[13,] 0.46    0 0.49 0.06
[14,] 0.24    0 0.20 0.56</code></pre>
<pre class="r"><code>diag(cor(fit.my$weight,fit.music$Weight.gene,use = &quot;complete.obs&quot;))</code></pre>
<pre><code> [1] 0.6086247 0.5920780 0.8431938 0.8497132 0.8188233 0.2454806 0.3132627
 [8] 0.8122491 0.2628207 0.4931705 0.7985513 0.5250052 0.8232666 0.8267241</code></pre>
<p>do not use weights when calculating variance</p>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.096 0.032 0.037 0.077
 [2,] 0.124 0.039 0.044 0.100
 [3,] 0.046 0.018 0.015 0.034
 [4,] 0.035 0.018 0.015 0.026
 [5,] 0.143 0.017 0.091 0.049
 [6,] 0.062 0.023 0.056 0.060
 [7,] 0.097 0.015 0.029 0.075
 [8,] 0.071 0.023 0.026 0.053
 [9,] 0.062 0.024 0.031 0.055
[10,] 0.080 0.020 0.041 0.083
[11,] 0.054 0.025 0.022 0.044
[12,] 0.039 0.018 0.019 0.031
[13,] 0.211 0.026 0.170 0.033
[14,] 0.052 0.016 0.016 0.054</code></pre>
<pre class="r"><code>is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE  TRUE
 [2,]  TRUE  TRUE FALSE  TRUE
 [3,]  TRUE FALSE FALSE FALSE
 [4,] FALSE FALSE FALSE FALSE
 [5,] FALSE FALSE  TRUE FALSE
 [6,]  TRUE FALSE FALSE FALSE
 [7,]  TRUE FALSE FALSE  TRUE
 [8,]  TRUE FALSE FALSE FALSE
 [9,]  TRUE FALSE  TRUE  TRUE
[10,] FALSE FALSE  TRUE  TRUE
[11,]  TRUE FALSE FALSE FALSE
[12,] FALSE FALSE FALSE FALSE
[13,] FALSE FALSE  TRUE FALSE
[14,] FALSE FALSE FALSE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.3214286</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc2&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.096 0.034 0.037 0.077
 [2,] 0.125 0.041 0.045 0.101
 [3,] 0.047 0.020 0.015 0.035
 [4,] 0.035 0.019 0.016 0.027
 [5,] 0.143 0.018 0.093 0.050
 [6,] 0.062 0.025 0.058 0.062
 [7,] 0.097 0.017 0.029 0.075
 [8,] 0.071 0.023 0.028 0.054
 [9,] 0.063 0.027 0.032 0.056
[10,] 0.080 0.022 0.042 0.085
[11,] 0.055 0.026 0.027 0.046
[12,] 0.039 0.020 0.020 0.032
[13,] 0.211 0.027 0.171 0.033
[14,] 0.052 0.017 0.018 0.055</code></pre>
<pre class="r"><code>is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE  TRUE
 [2,]  TRUE  TRUE FALSE  TRUE
 [3,]  TRUE FALSE FALSE FALSE
 [4,] FALSE FALSE FALSE  TRUE
 [5,] FALSE FALSE  TRUE FALSE
 [6,]  TRUE FALSE FALSE FALSE
 [7,]  TRUE FALSE FALSE  TRUE
 [8,]  TRUE FALSE FALSE FALSE
 [9,]  TRUE FALSE  TRUE  TRUE
[10,] FALSE FALSE  TRUE  TRUE
[11,]  TRUE FALSE FALSE FALSE
[12,] FALSE FALSE FALSE FALSE
[13,] FALSE FALSE  TRUE FALSE
[14,] FALSE FALSE FALSE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.3392857</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc3&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.097 0.035 0.037 0.078
 [2,] 0.125 0.043 0.046 0.102
 [3,] 0.047 0.022 0.016 0.036
 [4,] 0.036 0.021 0.017 0.028
 [5,] 0.145 0.019 0.096 0.051
 [6,] 0.063 0.026 0.061 0.064
 [7,] 0.097 0.018 0.029 0.076
 [8,] 0.071 0.023 0.032 0.055
 [9,] 0.063 0.031 0.033 0.057
[10,] 0.081 0.024 0.044 0.087
[11,] 0.056 0.028 0.035 0.049
[12,] 0.040 0.022 0.023 0.033
[13,] 0.212 0.030 0.172 0.033
[14,] 0.052 0.019 0.021 0.056</code></pre>
<pre class="r"><code>is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE  TRUE
 [2,]  TRUE  TRUE  TRUE  TRUE
 [3,]  TRUE FALSE FALSE FALSE
 [4,] FALSE FALSE  TRUE  TRUE
 [5,] FALSE FALSE  TRUE FALSE
 [6,]  TRUE FALSE FALSE FALSE
 [7,]  TRUE FALSE FALSE  TRUE
 [8,]  TRUE FALSE FALSE FALSE
 [9,]  TRUE FALSE  TRUE  TRUE
[10,] FALSE FALSE  TRUE  TRUE
[11,]  TRUE FALSE FALSE FALSE
[12,] FALSE FALSE FALSE FALSE
[13,] FALSE FALSE  TRUE FALSE
[14,] FALSE FALSE FALSE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,14))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.375</code></pre>
<p><strong>gene weights comparison with cibersort x</strong></p>
<pre class="r"><code>baron_sig_cibersort_output = read.delim(&quot;data/pancreas/baron_sig_cibersort_output.txt&quot;, quote=&quot;&quot;)
sig_genes = baron_sig_cibersort_output[,1]
length(sig_genes)</code></pre>
<pre><code>[1] 1359</code></pre>
<pre class="r"><code>genes_all = rownames(fit.music$Weight.gene)
w_cibersort = rep(0,length(genes_all))
names(w_cibersort) = genes_all
w_cibersort[names(w_cibersort)%in%sig_genes] = 1


cor(fit.my$weight[,1],w_cibersort,use=&#39;complete.obs&#39;)</code></pre>
<pre><code>[1] -0.04233405</code></pre>
<pre class="r"><code>cor(fit.music$Weight.gene[,1],w_cibersort,use=&#39;complete.obs&#39;)</code></pre>
<pre><code>[1] -0.1127277</code></pre>
<pre class="r"><code>cor(fit.my$weight[,2],w_cibersort,use=&#39;complete.obs&#39;)</code></pre>
<pre><code>[1] -0.03486967</code></pre>
<pre class="r"><code>cor(fit.music$Weight.gene[,2],w_cibersort,use=&#39;complete.obs&#39;)</code></pre>
<pre><code>[1] -0.09137873</code></pre>
<pre class="r"><code>cor(fit.my$weight[,3],w_cibersort,use=&#39;complete.obs&#39;)</code></pre>
<pre><code>[1] -0.03249847</code></pre>
<pre class="r"><code>cor(fit.music$Weight.gene[,3],w_cibersort,use=&#39;complete.obs&#39;)</code></pre>
<pre><code>[1] -0.0931649</code></pre>
<pre class="r"><code>#plot(w_cibersort,fit.music$Weight.gene[,1])
#cibersort result

baron_cibersort_Results = read.delim(&quot;data/pancreas/baron_cibersort_Results.txt&quot;, quote=&quot;&quot;)
cibersort_est = baron_cibersort_Results[,2:5]
cibersort_est = cibersort_est[,c(3,1,2,4)]
round(cibersort_est,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.24    0  0.27  0.49
2   0.18    0  0.28  0.53
3   0.22    0  0.20  0.59
4   0.20    0  0.27  0.53
5   0.47    0  0.34  0.18
6   0.24    0  0.45  0.31
7   0.22    0  0.21  0.57
8   0.14    0  0.21  0.64
9   0.17    0  0.29  0.54
10  0.33    0  0.30  0.36
11  0.26    0  0.17  0.57
12  0.38    0  0.22  0.40
13  0.40    0  0.51  0.09
14  0.25    0  0.11  0.64</code></pre>
<pre class="r"><code>rmse(as.matrix(cibersort_est),(rep(1,14))%*%t(b))</code></pre>
<pre><code>[1] 0.1458751</code></pre>
</div>
<div id="create-bulk-datawith-technical-noise-using-seger-dataset." class="section level2">
<h2>Create bulk data(with technical noise) using Seger dataset.</h2>
<pre class="r"><code>datax = set_data_decon(Y = seger,cell_types = cell_types, gene_thresh = 0,max_count_quantile = 1,w=1)
design.mat = scRef_multi_proc(datax$Y,datax$cell_type_idx,datax$indi_idx,estimator=&quot;separate&quot;,est_sigma2 = FALSE)
# create bulk data
b = c(0.1,0.1,0.3,0.5)
bulk.seger = createBulk_tech.noise(design.mat$X_array,b,gene_names = rownames(seger),seed=12345,ls_per_gene = 500)</code></pre>
</div>
<div id="xin-as-reference" class="section level2">
<h2>Xin as reference</h2>
<pre class="r"><code># fit music
xin.es = sce_to_es(xin_raw)
cell_size = data.frame(type = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;), size=c(1,1,1,1))
fit.music = music_prop(bulk_mat_to_es(bulk.seger$y),xin.es,clusters = &#39;cell_type&#39;, samples = &#39;individual&#39;,
                       select.ct = cell_types,cell_size=cell_size,verbose = FALSE,nu=1e-4,normalize = F)
rmse(fit.music$Est.prop.weighted,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.1197138</code></pre>
<pre class="r"><code>round(fit.music$Est.prop.weighted,2)</code></pre>
<pre><code>   alpha beta delta gamma
1   0.21 0.00  0.51  0.29
2   0.18 0.00  0.39  0.43
3   0.23 0.00  0.41  0.37
4   0.17 0.00  0.58  0.25
5   0.17 0.00  0.46  0.36
6   0.11 0.00  0.39  0.50
7   0.11 0.00  0.38  0.51
8   0.20 0.00  0.42  0.38
9   0.17 0.01  0.40  0.42
10  0.16 0.00  0.45  0.40</code></pre>
<pre class="r"><code>datax = set_data_decon(y = bulk.seger$y, Y = xin_raw[,-rm.indi.idx],
                       cell_types = cell_types, 
                       gene_thresh = 0,max_count_quantile = 1,w=1,filter.gene = F)

design.mat = scRef_multi_proc(datax$Y,datax$cell_type_idx,datax$indi_idx,
                                    estimator=&quot;separate&quot;,est_sigma2 = FALSE)

fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
rmse(fit.my$beta_hat,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.08526634</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.04 0.01 0.46 0.49
 [2,] 0.14 0.01 0.33 0.51
 [3,] 0.06 0.01 0.56 0.38
 [4,] 0.05 0.01 0.43 0.51
 [5,] 0.11 0.02 0.30 0.57
 [6,] 0.03 0.01 0.38 0.58
 [7,] 0.08 0.03 0.30 0.59
 [8,] 0.19 0.02 0.22 0.58
 [9,] 0.17 0.02 0.23 0.58
[10,] 0.05 0.02 0.32 0.61</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.010 0.005 0.066 0.065
 [2,] 0.011 0.004 0.035 0.036
 [3,] 0.011 0.003 0.052 0.051
 [4,] 0.008 0.003 0.071 0.071
 [5,] 0.009 0.007 0.050 0.050
 [6,] 0.011 0.005 0.042 0.042
 [7,] 0.007 0.006 0.033 0.033
 [8,] 0.009 0.005 0.037 0.038
 [9,] 0.010 0.008 0.033 0.034
[10,] 0.011 0.007 0.054 0.055</code></pre>
<pre class="r"><code>is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] FALSE FALSE FALSE  TRUE
 [2,] FALSE FALSE  TRUE  TRUE
 [3,] FALSE FALSE FALSE FALSE
 [4,] FALSE FALSE  TRUE  TRUE
 [5,]  TRUE FALSE  TRUE  TRUE
 [6,] FALSE FALSE  TRUE  TRUE
 [7,] FALSE FALSE  TRUE FALSE
 [8,] FALSE FALSE FALSE FALSE
 [9,] FALSE FALSE FALSE FALSE
[10,] FALSE FALSE  TRUE FALSE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.3</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc2&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
rmse(fit.my$beta_hat,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.08526634</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.04 0.01 0.46 0.49
 [2,] 0.14 0.01 0.33 0.51
 [3,] 0.06 0.01 0.56 0.38
 [4,] 0.05 0.01 0.43 0.51
 [5,] 0.11 0.02 0.30 0.57
 [6,] 0.03 0.01 0.38 0.58
 [7,] 0.08 0.03 0.30 0.59
 [8,] 0.19 0.02 0.22 0.58
 [9,] 0.17 0.02 0.23 0.58
[10,] 0.05 0.02 0.32 0.61</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>        [,1]  [,2]   [,3]  [,4]
 [1,] 10.124 1.515 11.147 2.539
 [2,]  0.012 0.008  0.036 0.037
 [3,]  0.012 0.004  0.052 0.052
 [4,]  0.008 0.004  0.072 0.072
 [5,]  0.010 0.010  0.050 0.051
 [6,]  7.516 1.739  3.802 1.976
 [7,]  0.030 0.056  0.059 0.040
 [8,]  0.010 0.005  0.037 0.038
 [9,]  0.012 0.012  0.034 0.034
[10,]  0.039 0.011  0.055 0.058</code></pre>
<pre class="r"><code>is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,]  TRUE  TRUE  TRUE  TRUE
 [2,] FALSE FALSE  TRUE  TRUE
 [3,] FALSE FALSE FALSE FALSE
 [4,] FALSE FALSE  TRUE  TRUE
 [5,]  TRUE FALSE  TRUE  TRUE
 [6,]  TRUE  TRUE  TRUE  TRUE
 [7,]  TRUE  TRUE  TRUE FALSE
 [8,] FALSE FALSE FALSE FALSE
 [9,] FALSE FALSE FALSE FALSE
[10,]  TRUE FALSE  TRUE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.525</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc3&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = T)
rmse(fit.my$beta_hat,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.08526634</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.04 0.01 0.46 0.49
 [2,] 0.14 0.01 0.33 0.51
 [3,] 0.06 0.01 0.56 0.38
 [4,] 0.05 0.01 0.43 0.51
 [5,] 0.11 0.02 0.30 0.57
 [6,] 0.03 0.01 0.38 0.58
 [7,] 0.08 0.03 0.30 0.59
 [8,] 0.19 0.02 0.22 0.58
 [9,] 0.17 0.02 0.23 0.58
[10,] 0.05 0.02 0.32 0.61</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>          [,1]    [,2]     [,3]    [,4]
 [1,] 1342.538 200.972 1478.222 336.655
 [2,]    0.015   0.015    0.038   0.038
 [3,]    0.013   0.006    0.053   0.052
 [4,]    0.009   0.005    0.072   0.072
 [5,]    0.012   0.013    0.050   0.051
 [6,]  996.953 230.670  504.309 261.974
 [7,]    0.129   0.241    0.214   0.104
 [8,]    0.010   0.007    0.037   0.039
 [9,]    0.014   0.018    0.035   0.035
[10,]    0.125   0.027    0.065   0.084</code></pre>
<pre class="r"><code>is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,]  TRUE  TRUE  TRUE  TRUE
 [2,] FALSE FALSE  TRUE  TRUE
 [3,] FALSE FALSE FALSE FALSE
 [4,] FALSE FALSE  TRUE  TRUE
 [5,]  TRUE FALSE  TRUE  TRUE
 [6,]  TRUE  TRUE  TRUE  TRUE
 [7,]  TRUE  TRUE  TRUE  TRUE
 [8,] FALSE FALSE FALSE FALSE
 [9,] FALSE FALSE FALSE FALSE
[10,]  TRUE FALSE  TRUE  TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 0.55</code></pre>
<pre class="r"><code>ii = which(rownames(fit.music$Weight.gene)%in%rownames(fit.my$weight))
diag(cor(fit.my$weight,fit.music$Weight.gene[ii,],use = &quot;complete.obs&quot;))</code></pre>
<pre><code> [1] 0.11847550 0.15119127 0.09649927 0.12203838 0.11564573 0.16593279
 [7] 0.13326481 0.13345121 0.10939221 0.07143041</code></pre>
<pre class="r"><code>diag(cor(fit.my$X,fit.music$X[ii,],use = &quot;pairwise.complete.obs&quot;))</code></pre>
<pre><code>    alpha      beta     delta     gamma 
0.9990868 0.9985348 0.9999121 0.9952195 </code></pre>
<pre class="r"><code>diag(cor(fit.my$Vg[,c(1,6,11,16)],fit.music$Sigma[ii,],use = &quot;pairwise.complete.obs&quot;))</code></pre>
<pre><code>[1] 0.9962446 0.9811555 0.9995698 0.9956617</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc0&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
rmse(fit.my$beta_hat,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.08526634</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.04 0.01 0.46 0.49
 [2,] 0.14 0.01 0.33 0.51
 [3,] 0.06 0.01 0.56 0.38
 [4,] 0.05 0.01 0.43 0.51
 [5,] 0.11 0.02 0.30 0.57
 [6,] 0.03 0.01 0.38 0.58
 [7,] 0.08 0.03 0.30 0.59
 [8,] 0.19 0.02 0.22 0.58
 [9,] 0.17 0.02 0.23 0.58
[10,] 0.05 0.02 0.32 0.61</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.103 1.682 0.508 1.351
 [2,] 0.068 0.537 0.163 0.406
 [3,] 0.093 1.448 0.405 1.147
 [4,] 0.104 1.782 0.530 1.445
 [5,] 0.077 0.535 0.355 0.294
 [6,] 0.060 0.496 0.113 0.439
 [7,] 0.057 0.744 0.194 0.562
 [8,] 0.139 0.592 0.267 0.300
 [9,] 0.115 0.476 0.201 0.249
[10,] 0.070 0.827 0.363 0.624</code></pre>
<pre class="r"><code>is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] TRUE TRUE TRUE TRUE
 [2,] TRUE TRUE TRUE TRUE
 [3,] TRUE TRUE TRUE TRUE
 [4,] TRUE TRUE TRUE TRUE
 [5,] TRUE TRUE TRUE TRUE
 [6,] TRUE TRUE TRUE TRUE
 [7,] TRUE TRUE TRUE TRUE
 [8,] TRUE TRUE TRUE TRUE
 [9,] TRUE TRUE TRUE TRUE
[10,] TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc2&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
rmse(fit.my$beta_hat,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.08526634</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.04 0.01 0.46 0.49
 [2,] 0.14 0.01 0.33 0.51
 [3,] 0.06 0.01 0.56 0.38
 [4,] 0.05 0.01 0.43 0.51
 [5,] 0.11 0.02 0.30 0.57
 [6,] 0.03 0.01 0.38 0.58
 [7,] 0.08 0.03 0.30 0.59
 [8,] 0.19 0.02 0.22 0.58
 [9,] 0.17 0.02 0.23 0.58
[10,] 0.05 0.02 0.32 0.61</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.103 1.690 0.512 1.358
 [2,] 0.068 0.540 0.165 0.409
 [3,] 0.093 1.454 0.406 1.153
 [4,] 0.105 1.792 0.536 1.453
 [5,] 0.077 0.539 0.362 0.296
 [6,] 0.060 0.500 0.116 0.442
 [7,] 0.057 0.746 0.194 0.563
 [8,] 0.139 0.593 0.271 0.300
 [9,] 0.115 0.478 0.204 0.249
[10,] 0.070 0.833 0.370 0.628</code></pre>
<pre class="r"><code>is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] TRUE TRUE TRUE TRUE
 [2,] TRUE TRUE TRUE TRUE
 [3,] TRUE TRUE TRUE TRUE
 [4,] TRUE TRUE TRUE TRUE
 [5,] TRUE TRUE TRUE TRUE
 [6,] TRUE TRUE TRUE TRUE
 [7,] TRUE TRUE TRUE TRUE
 [8,] TRUE TRUE TRUE TRUE
 [9,] TRUE TRUE TRUE TRUE
[10,] TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>fit.my = wols(design.mat$X,datax$y,design.mat$Vg,b=NULL,
              w.mode = &#39;res+ref_var&#39;,adj.v = T,X_array = design.mat$X_array,
              hc.type = &#39;hc3&#39;,marker = rownames(fit.music$Weight.gene),
              nu=1e-4,scale.y = F,use.weight.for.var = F)
rmse(fit.my$beta_hat,(rep(1,10))%*%t(b))</code></pre>
<pre><code>[1] 0.08526634</code></pre>
<pre class="r"><code>round(fit.my$beta_hat,2)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] 0.04 0.01 0.46 0.49
 [2,] 0.14 0.01 0.33 0.51
 [3,] 0.06 0.01 0.56 0.38
 [4,] 0.05 0.01 0.43 0.51
 [5,] 0.11 0.02 0.30 0.57
 [6,] 0.03 0.01 0.38 0.58
 [7,] 0.08 0.03 0.30 0.59
 [8,] 0.19 0.02 0.22 0.58
 [9,] 0.17 0.02 0.23 0.58
[10,] 0.05 0.02 0.32 0.61</code></pre>
<pre class="r"><code>round(fit.my$beta_hat_se,3)</code></pre>
<pre><code>       [,1]  [,2]  [,3]  [,4]
 [1,] 0.104 1.699 0.516 1.366
 [2,] 0.068 0.543 0.168 0.411
 [3,] 0.094 1.461 0.407 1.158
 [4,] 0.105 1.802 0.542 1.461
 [5,] 0.077 0.543 0.369 0.297
 [6,] 0.060 0.503 0.118 0.445
 [7,] 0.057 0.748 0.194 0.565
 [8,] 0.140 0.595 0.276 0.301
 [9,] 0.116 0.479 0.208 0.250
[10,] 0.070 0.838 0.377 0.632</code></pre>
<pre class="r"><code>is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se)</code></pre>
<pre><code>      [,1] [,2] [,3] [,4]
 [1,] TRUE TRUE TRUE TRUE
 [2,] TRUE TRUE TRUE TRUE
 [3,] TRUE TRUE TRUE TRUE
 [4,] TRUE TRUE TRUE TRUE
 [5,] TRUE TRUE TRUE TRUE
 [6,] TRUE TRUE TRUE TRUE
 [7,] TRUE TRUE TRUE TRUE
 [8,] TRUE TRUE TRUE TRUE
 [9,] TRUE TRUE TRUE TRUE
[10,] TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code>mean(is.covered((rep(1,10))%*%t(b),fit.my$beta_hat,fit.my$beta_hat_se))</code></pre>
<pre><code>[1] 1</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
 [3] GenomicRanges_1.42.0        GenomeInfoDb_1.26.1        
 [5] MatrixGenerics_1.2.0        matrixStats_0.57.0         
 [7] xbioc_0.1.19                AnnotationDbi_1.52.0       
 [9] IRanges_2.24.0              S4Vectors_0.28.0           
[11] Biobase_2.50.0              BiocGenerics_0.36.0        
[13] MuSiC_0.1.1                 ggplot2_3.3.2              
[15] nnls_1.4                    workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] bit64_4.0.5            assertthat_0.2.1       BiocManager_1.30.10   
 [4] blob_1.2.1             GenomeInfoDbData_1.2.4 yaml_2.2.1            
 [7] pillar_1.4.6           RSQLite_2.2.1          backports_1.1.10      
[10] lattice_0.20-41        quantreg_5.75          glue_1.4.2            
[13] digest_0.6.27          promises_1.1.1         XVector_0.30.0        
[16] checkmate_2.0.0        colorspace_1.4-1       htmltools_0.5.1.1     
[19] httpuv_1.5.4           Matrix_1.2-18          plyr_1.8.6            
[22] conquer_1.0.2          pkgconfig_2.0.3        SparseM_1.78          
[25] zlibbioc_1.36.0        purrr_0.3.4            xtable_1.8-4          
[28] scales_1.1.1           whisker_0.4            later_1.1.0.1         
[31] MatrixModels_0.4-1     git2r_0.27.1           tibble_3.0.4          
[34] pkgmaker_0.32.2.900    generics_0.1.0         ellipsis_0.3.1        
[37] withr_2.3.0            magrittr_2.0.1         crayon_1.3.4          
[40] memoise_1.1.0          mcmc_0.9-7             evaluate_0.14         
[43] fs_1.5.0               MASS_7.3-53            tools_4.0.3           
[46] registry_0.5-1         lifecycle_1.0.0        stringr_1.4.0         
[49] MCMCpack_1.4-9         munsell_0.5.0          DelayedArray_0.16.0   
[52] compiler_4.0.3         rlang_0.4.10           grid_4.0.3            
[55] RCurl_1.98-1.2         rstudioapi_0.11        bitops_1.0-6          
[58] rmarkdown_2.5          gtable_0.3.0           codetools_0.2-16      
[61] DBI_1.1.0              reshape2_1.4.4         R6_2.4.1              
[64] knitr_1.30             dplyr_1.0.5            bit_4.0.4             
[67] rprojroot_1.3-2        stringi_1.5.3          Rcpp_1.0.5            
[70] vctrs_0.3.7            tidyselect_1.1.0       xfun_0.18             
[73] coda_0.19-4           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
