<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="DongyueXie" />

<meta name="date" content="2020-04-30" />

<title>Walk through MuSiC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">deconference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="about.html">Data</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/DongyueXie/deconference">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Walk through MuSiC</h1>
<h4 class="author"><em>DongyueXie</em></h4>
<h4 class="date"><em>2020-04-30</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>deconference/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200403code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200403)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200403code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200403)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/deconference/data/MuSiC/GSE50244bulkeset.rds
</td>
<td style="text-align:left;">
data/MuSiC/GSE50244bulkeset.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
~/deconference/data/MuSiC/EMTABesethealthy.rds
</td>
<td style="text-align:left;">
data/MuSiC/EMTABesethealthy.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetreeb45300c3032545b969e162f0198bd093ad76424dtargetblankb45300ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DongyueXie/deconference/tree/b45300c3032545b969e162f0198bd093ad76424d" target="_blank">b45300c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetreeb45300c3032545b969e162f0198bd093ad76424dtargetblankb45300ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  code/bulkSimulationFake.R
    Untracked:  code/deconference.R
    Untracked:  code/plot_simu.R
    Untracked:  code/scSimulation.R
    Untracked:  code/scSimulationFake.R
    Untracked:  data/MuSiC/
    Untracked:  output/bulkref_pbmc_allgene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls5_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls5_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls05_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls10_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls50_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls10_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls05_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_addw_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_addw_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN30_bulkls50_addw_tauunknown.RData

Unstaged changes:
    Modified:   code/gls.R
    Deleted:    code/plot_bulksimu.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/b45300c3032545b969e162f0198bd093ad76424d/analysis/MuSiC.Rmd" target="_blank">b45300c</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-04-30
</td>
<td>
wflow_publish(“analysis/MuSiC.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<span class="math display">\[\begin{array}{*{20}{l}} {\mathrm{Var}\left[ {Y_{jg}|{\boldsymbol{p}}_j} \right]} \hfill &amp; = \hfill &amp; {E\left[ {\mathrm{Var}\left[ {Y_{jg}{\mathrm{|}}{\boldsymbol{p}}_j,\,{\boldsymbol{\theta }}_{jg}} \right]} \right] + \mathrm{Var}\left[ {E\left[ {Y_{jg}|{\boldsymbol{p}}_j,\,{\boldsymbol{\theta }}_{jg}} \right]} \right]} \hfill \\ {} \hfill &amp; = \hfill &amp; {C_j^2\delta _{jg}^2 + \mathrm{Var}\left[ {C_j \cdot \mathop {\sum }\limits_{k = 1}^K p_{jk}\,S_k\,\theta _{jg}^k} \right]} \hfill \\ {} \hfill &amp; = \hfill &amp; {C_j^2\delta _{jg}^2 + C_j^2 \cdot \mathop {\sum }\limits_{k = 1}^K p_{jk}^2\,S_k^2\,\mathrm{Var}\left[ {\theta _{jg}^k} \right] = C_j^2\delta _{jg}^2 + C_j^2\mathop {\sum }\limits_{k = 1}^K p_{jk}^2S_k^2\,\sigma _{gk}^2} \hfill \\ {} \hfill &amp; = \hfill &amp; 1/w_{jg} \hfill \end{array}\]</span>
<p><span class="math inline">\(j\)</span> indexes individual and <span class="math inline">\(g\)</span> indexes gene.</p>
</div>
<div id="walk-through" class="section level1">
<h1>Walk through</h1>
<pre class="r"><code>library(MuSiC)
GSE50244.bulk.eset &lt;- readRDS(&quot;~/deconference/data/MuSiC/GSE50244bulkeset.rds&quot;)
EMTAB.eset &lt;- readRDS(&quot;~/deconference/data/MuSiC/EMTABesethealthy.rds&quot;)
library(xbioc)</code></pre>
<p>6 major cell types: alpha, beta, delta, gamma, acinar and ductal, which make up over 90% of the whole islet.</p>
<p>set data and input parameters</p>
<pre class="r"><code>bulk.eset = GSE50244.bulk.eset
sc.eset = EMTAB.eset
clusters = &#39;cellType&#39;
samples = &#39;sampleID&#39;
select.ct = c(&#39;alpha&#39;, &#39;beta&#39;, &#39;delta&#39;, &#39;gamma&#39;,&#39;acinar&#39;, &#39;ductal&#39;)
verbose = F


# filter out genes has no counts in bulk data
bulk.gene = rownames(bulk.eset)[rowMeans(exprs(bulk.eset)) != 0]
bulk.eset = bulk.eset[bulk.gene, , drop = FALSE]</code></pre>
<div id="process-single-cell-reference-data" class="section level2">
<h2>Process single cell reference data</h2>
<p>extract single cell information, function: <code>music_basis</code>, This function is used for generating cell type specific cross-subject mean and variance for each gene. Cell type specific library size is also calcualted.</p>
<pre class="r"><code>sc.markers = bulk.gene
x = sc.eset; non.zero = TRUE; markers = sc.markers; clusters = clusters; samples = samples; select.ct = select.ct; cell_size = NULL; ct.cov = NULL; verbose = verbose

# select cell type in sc data
s.ct = sampleNames(x)[as.character(pVar(x, clusters)) %in% select.ct]
x &lt;- x[, s.ct, drop = FALSE]

## eliminate non expressed genes
nz.gene = rownames(x)[( rowSums(exprs(x)) != 0 )]
x &lt;- x[nz.gene, , drop = FALSE]

clusters &lt;- as.character(pVar(x, clusters))
samples &lt;- as.character(pVar(x, samples))

# 6 individuals, total 995 cells.

M.theta &lt;- sapply(unique(clusters), function(ct){
    my.rowMeans(sapply(unique(samples), function(sid){
      y = exprs(x)[,clusters %in% ct &amp; samples %in% sid, drop = FALSE]
      rowSums(y)/sum(y)
    }), na.rm = TRUE)
  })
dim(M.theta)</code></pre>
<pre><code>[1] 21464     6</code></pre>
<p>M.theta is a gene by cell type matrix. Each column is gene relative expression in a cell type. How M.theta is calculated? For each cell type <span class="math inline">\(k\)</span>, calculate gene relative expression in each individual by rowSums(y)/sum(y) (note: not mean of every cell, but aggregate all cells), then take the mean of gene relative expression in each individual as <span class="math inline">\(\hat x_{k}\)</span>(a vector, of length the number of genes). In short, <span class="math inline">\(\hat x_k = \frac{1}{N_{individual}}\sum_i \hat x_{ik}\)</span>, where <span class="math inline">\(\hat x_{ik} = \frac{rowSums(Y_{ik})}{sum(Y_{ik})}\)</span> and <span class="math inline">\(Y_{ik}\)</span> is gene by cell matrix of individual <span class="math inline">\(i\)</span> and cell type <span class="math inline">\(k\)</span>.</p>
<p>Next step is to estimate variance across individuals.</p>
<pre class="r"><code>Sigma &lt;- sapply(unique(clusters), function(ct){
      apply(sapply(unique(samples), function(sid){
        y = exprs(x)[,clusters %in% ct &amp; samples %in% sid, drop = FALSE]
        rowSums(y)/sum(y)
      }), 1, var, na.rm = TRUE)
    })
dim(Sigma)</code></pre>
<pre><code>[1] 21464     6</code></pre>
<pre class="r"><code>m.ct = match(select.ct, colnames(Sigma))
Sigma = Sigma[, m.ct]</code></pre>
<p>Sigma is a gene by cell matrix. Its entry <span class="math inline">\(v_{gk}\)</span> is an estimate of gene relative expression in cell type <span class="math inline">\(k\)</span> variance across individuals. How? For each cell type <span class="math inline">\(k\)</span>, calculate sample variance of <span class="math inline">\(\hat x_{ik}\)</span> for <span class="math inline">\(i=1,2,...,6\)</span>.</p>
<p>MuSiC has another option of setting ct.cov = TRUE, saying “If TRUE, use the covariance across cell types”. Let’s see what it does.</p>
<pre class="r"><code>    nGenes = nrow(x);
    n.ct = length(unique(clusters));
    nSubs = length(unique(samples))
    
    Theta &lt;- sapply(unique(clusters), function(ct){
      sapply(unique(samples), function(sid){
        y = exprs(x)[,clusters %in% ct &amp; samples %in% sid, drop = FALSE]
        return( rowSums(y)/sum(y) )
      })
    })
    if(!is.null(select.ct)){
      m.ct = match(select.ct, colnames(Theta))
      Theta = Theta[, m.ct]
    }
    
    Sigma.ct = sapply(1:nGenes, function(g){
      sigma.temp = Theta[nGenes*(0:(nSubs - 1)) + g, ];
      Cov.temp = cov(sigma.temp)
      Cov.temp1 = cov(sigma.temp[rowSums(is.na(Theta[nGenes*(0:(nSubs - 1)) + 1, ])) == 0, ])
      Cov.temp[which(colSums(is.na(sigma.temp))&gt;0), ] = Cov.temp1[which(colSums(is.na(sigma.temp))&gt;0), ]
      Cov.temp[, which(colSums(is.na(sigma.temp))&gt;0)] = Cov.temp1[, which(colSums(is.na(sigma.temp))&gt;0)]
      return(Cov.temp)
    })
    colnames(Sigma.ct) = rownames(x)</code></pre>
<p>Sigma.ct is <span class="math inline">\((N_{individual}*N_{cell Type})\times N_{gene}\)</span> matrix. Each column is a vectorized covaraince matrix, which is gene expression cov among cell types, obtained using <span class="math inline">\(cov(X_{gik})\)</span>. <span class="math inline">\(X_{gik}\)</span> is a <span class="math inline">\(N_{individual}*N_{cell Type}\)</span> matrix of gene <span class="math inline">\(g\)</span>’s relative expresison.</p>
<p>So the difference between Sigma and Sigma.ct is that Sigma ignores the covariance among cell types.</p>
<p>The final step is to calculate cell size: for each cell type <span class="math inline">\(k\)</span>, take the average of <span class="math inline">\(S_{ik}\)</span> for <span class="math inline">\(i=1,2,..,6\)</span> where <span class="math inline">\(S_{ik} = \frac{sum(Y_{ik})}{ncol(Y_{ik})}\)</span>.</p>
<pre class="r"><code> S &lt;- sapply(unique(clusters), function(ct){
    my.rowMeans(sapply(unique(samples), function(sid){
      y = exprs(x)[, clusters %in% ct &amp; samples %in% sid, drop = FALSE]
      sum(y)/ncol(y)
    }), na.rm = TRUE)
  })
S</code></pre>
<pre><code>     delta     alpha     gamma    ductal    acinar     beta
1 771492.5 1112535.6 441998.45  745700.5 1931490.1 672700.3
2 813057.3  950452.0 656086.57 1065658.7 1082554.4 843435.9
4 328635.8  384778.1 320486.22  443171.8  540339.3 362149.6
5 318821.2  263706.3 270594.95  354052.4  505857.6 328880.0
6 250652.9  240645.2  87412.08  371934.3  241888.4 137893.2
8 119419.0  231126.2 506790.55  192096.1  112532.5 139679.5</code></pre>
<pre class="r"><code>M.S = colMeans(S, na.rm = TRUE)
M.S</code></pre>
<pre><code>   delta    alpha    gamma   ductal   acinar     beta 
433679.8 530540.6 380561.5 528769.0 735777.1 414123.1 </code></pre>
<p>Obtain the design matrix for regression. Note that MuSiC estimates cell type proportion with strict assumptions. So the design matirx is relative expression times cell size.</p>
<pre class="r"><code>D &lt;- t(t(M.theta)*M.S)
sc.basis = list(Disgn.mtx = D, S = S, M.S = M.S, M.theta = M.theta, Sigma = Sigma)</code></pre>
</div>
<div id="perform-regression" class="section level2">
<h2>Perform regression</h2>
<p>Note: MuSiC selets common genes among bulk and reference data after processing reference data.</p>
<p>Let’s take a look at how it handles ct.cov=TRUE.</p>
<pre class="r"><code>sc.basis = list(Disgn.mtx = D, S = S, M.S = M.S, M.theta = M.theta, Sigma.ct = Sigma.ct)
cm.gene = intersect( rownames(sc.basis$Disgn.mtx), bulk.gene )
m.sc = match(cm.gene, rownames(sc.basis$Disgn.mtx))
m.bulk = match(cm.gene, bulk.gene)
D1 = sc.basis$Disgn.mtx[m.sc, ] 
M.S = colMeans(sc.basis$S, na.rm = T)

## obtain bulk data gene relative expression
Yjg = relative.ab(exprs(bulk.eset)[m.bulk, ]); N.bulk = ncol(bulk.eset)

Sigma.ct = sc.basis$Sigma.ct[, m.sc];

i=1
## perform weighted ls ......

if(sum(Yjg[, i] == 0) &gt; 0){
      D1.temp = D1[Yjg[, i]!=0, ];
      Yjg.temp = Yjg[Yjg[, i]!=0, i];
      Sigma.ct.temp = Sigma.ct[, Yjg[,i]!=0];
      if(verbose) message(paste(colnames(Yjg)[i], &#39;has common genes&#39;, sum(Yjg[, i] != 0), &#39;...&#39;) )
    }else{
      D1.temp = D1;
      Yjg.temp = Yjg[, i];
      Sigma.ct.temp = Sigma.ct;
      if(verbose) message(paste(colnames(Yjg)[i], &#39;has common genes&#39;, sum(Yjg[, i] != 0), &#39;...&#39;))
    }</code></pre>
<p>Then Yjg.temp, D1.temp, M.S, Sigma.ct.temp are passed to function <code>music.iter.ct</code>. In this function there is a line. So I guess in order to avoid numerical issue, they multiply Y by 100.</p>
<pre class="r"><code>if(normalize){
    X = X/sd(as.vector(X));
    S = S*sd(as.vector(X));
    Y = Y/sd(Y)
  }else{
    Y = Y*100
  }</code></pre>
<p>The final function performing regression is <code>music.basic.ct</code>. The <span class="math inline">\(Var(E(y_{jg}|p_j,\theta_{jg}))\)</span> part in the weight of gene <span class="math inline">\(g\)</span> is calculated as <span class="math inline">\((Sp_j)^TVar(\theta_{jg})(Sp_j)\)</span>.</p>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Assume there are <span class="math inline">\(G\)</span> genes, <span class="math inline">\(K\)</span> cell types and <span class="math inline">\(I\)</span> individuals. Let <span class="math inline">\(Y_{ik}\)</span> denote the single cell reference matrix of individual <span class="math inline">\(i\)</span> and cell type <span class="math inline">\(k\)</span>.</p>
<p>MuSiC formulates the <span class="math inline">\(k\)</span>th column of reference matrix / signature matrix, <span class="math inline">\(\mathbf{x}_k\)</span> using the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>For each individual <span class="math inline">\(i=1,2,...,I\)</span>, calculate <span class="math inline">\(\hat{\mathbf{x}}_{ik} = \frac{rowSums(Y_{ik})}{sum(Y_{ik})}\)</span>.</p></li>
<li><p>Estimate <span class="math inline">\(\mathbf{x}_k\)</span> as <span class="math inline">\(\frac{1}{I}\sum_i \hat{\mathbf{x}}_{ik}\)</span>.</p></li>
</ol>
<p>MuSiC estimates the variance <span class="math inline">\(v_{gk} = Var(\hat{\mathbf{x}}_{gk})\)</span> as sample variance of <span class="math inline">\(\hat{\mathbf{x}}_{igk}\)</span>, <span class="math inline">\(i=1,2,...,I\)</span>.</p>
<p>Or, it estimates the covariance matrix <span class="math inline">\(V_{g} = Var(\hat{\mathbf{x}}_{g\cdot})\in R_+^{K\times K}\)</span> as sample covariance matrix of <span class="math inline">\(\hat{\mathbf{x}}_{ig\cdot}\)</span>, <span class="math inline">\(i=1,2,...,I\)</span>.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] xbioc_0.1.18         AnnotationDbi_1.44.0 IRanges_2.16.0      
[4] S4Vectors_0.20.1     Biobase_2.42.0       BiocGenerics_0.28.0 
[7] MuSiC_0.1.1          ggplot2_3.1.1        nnls_1.4            

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6        lattice_0.20-38     assertthat_0.2.0   
 [4] rprojroot_1.3-2     digest_0.6.18       R6_2.3.0           
 [7] plyr_1.8.4          backports_1.1.2     MatrixModels_0.4-1 
[10] RSQLite_2.1.1       evaluate_0.12       coda_0.19-2        
[13] highr_0.7           pillar_1.3.1        rlang_0.4.0        
[16] lazyeval_0.2.1      SparseM_1.78        whisker_0.3-2      
[19] blob_1.1.1          Matrix_1.2-15       rmarkdown_1.10     
[22] stringr_1.3.1       bit_1.1-14          munsell_0.5.0      
[25] compiler_3.5.1      httpuv_1.4.5        pkgconfig_2.0.2    
[28] pkgmaker_0.27       mcmc_0.9-7          htmltools_0.3.6    
[31] tidyselect_0.2.5    tibble_2.1.1        workflowr_1.6.0    
[34] codetools_0.2-15    crayon_1.3.4        dplyr_0.8.0.1      
[37] withr_2.1.2         later_0.7.5         MASS_7.3-51.1      
[40] grid_3.5.1          xtable_1.8-3        gtable_0.2.0       
[43] registry_0.5        DBI_1.0.0           git2r_0.26.1       
[46] magrittr_1.5        scales_1.0.0        bibtex_0.4.2       
[49] stringi_1.2.4       reshape2_1.4.3      fs_1.3.1           
[52] promises_1.0.1      tools_3.5.1         bit64_0.9-7        
[55] glue_1.3.0          purrr_0.3.2         yaml_2.2.0         
[58] colorspace_1.3-2    BiocManager_1.30.10 memoise_1.1.0      
[61] knitr_1.20          quantreg_5.36       MCMCpack_1.4-4     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
