<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="DongyueXie" />

<meta name="date" content="2020-10-29" />

<title>music gene length</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">deconference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="about.html">Data</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/DongyueXie/deconference">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">music gene length</h1>
<h4 class="author">DongyueXie</h4>
<h4 class="date">2020-10-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>deconference/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200403code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200403)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200403code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200403)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetreee366a21db9321aa742c44a6a6f929648c9d09b64targetblanke366a21a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DongyueXie/deconference/tree/e366a21db9321aa742c44a6a6f929648c9d09b64" target="_blank">e366a21</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDongyueXiedeconferencetreee366a21db9321aa742c44a6a6f929648c9d09b64targetblanke366a21a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DongyueXie/deconference/tree/e366a21db9321aa742c44a6a6f929648c9d09b64" target="_blank">e366a21</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/check_est_simu.Rmd
    Untracked:  analysis/compare_V_diag.Rmd
    Untracked:  analysis/multiIndNormal.Rmd
    Untracked:  analysis/multiref_seger.Rmd
    Untracked:  analysis/multiref_simu_zeroinf.Rmd
    Untracked:  analysis/multiref_simu_zeroinf_nondiag.Rmd
    Untracked:  analysis/plotGeneVar.Rmd
    Untracked:  code/mvmeta.R
    Untracked:  code/normal.R
    Untracked:  code/simu_func_multibulk_multiref.R
    Untracked:  data/EMTAB5061Segerstolpe/
    Untracked:  data/GSE129788/
    Untracked:  data/GSE141115/
    Untracked:  data/GSE81547Enge/
    Untracked:  data/GSE81608Xin/
    Untracked:  data/GSE84133Baron/
    Untracked:  data/GSE85241Muraro/
    Untracked:  data/MuSiC/
    Untracked:  data/all_genes_location.txt
    Untracked:  data/cibersort/
    Untracked:  data/gene_length.rds
    Untracked:  data/pancreas/
    Untracked:  data/pbmc3k/
    Untracked:  data/toSCE.R
    Untracked:  output/Xin_bycelltype_hc2.RData
    Untracked:  output/Xin_bycelltype_hc3.RData
    Untracked:  output/Xin_sampleM_hc2.RData
    Untracked:  output/Xin_sampleM_hc3.RData
    Untracked:  output/Xin_sampleM_unadj.RData
    Untracked:  output/Xin_universal_hc0.RData
    Untracked:  output/Xin_universal_hc2.RData
    Untracked:  output/Xin_universal_hc3.RData
    Untracked:  output/bulkref_pbmc_allgene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_allgene_refls5_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_fewgene_refls5_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls05_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls10_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls1_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls3_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls50_bulkls50.RData
    Untracked:  output/bulkref_pbmc_markergene_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls05_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls10_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls50_bulkls50.RData
    Untracked:  output/bulkref_simu_G1000_K4_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls10_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_addw_G1000_K4_refls5_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls05_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_addw_G1000_K4_refls1_bulkls50.RData
    Untracked:  output/bulkref_simu_correction_addw_G1000_K4_refls3_bulkls50.RData
    Untracked:  output/multiref_all.RData
    Untracked:  output/multiref_baron.RData
    Untracked:  output/multiref_enge.RData
    Untracked:  output/multiref_muraro.RData
    Untracked:  output/multiref_seger.RData
    Untracked:  output/multiref_test1.RData
    Untracked:  output/multiref_test2.RData
    Untracked:  output/multiref_test21.RData
    Untracked:  output/multiref_test22.RData
    Untracked:  output/multiref_test_1UMI.RData
    Untracked:  output/multiref_test_1UMI1nonUMI.RData
    Untracked:  output/multiref_test_2UMI.RData
    Untracked:  output/multiref_test_2UMI1nonUMI.RData
    Untracked:  output/multiref_test_2UMI2nonUMI.RData
    Untracked:  output/results_nsclc.RData
    Untracked:  output/scref_multiInd100_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_multiInd100_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown_Nosigma2.RData
    Untracked:  output/scref_multiInd10_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_multiInd10_simu_G500_K4_scls02_scN100_bulkls50_addw_tauunknown_Nosigma2.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls01_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls02_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN10_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN10_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN300_bulkls50_addw_tauunknown.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN30_bulkls50.RData
    Untracked:  output/scref_simu_G1000_K4_scls05_scN30_bulkls50_addw_tauunknown.RData
    Untracked:  output/twoBulk_bulkref_simu_G1000_K4_refls30.RData
    Untracked:  output/twoBulk_bulkref_simu_G500_K4_refls30.RData
    Untracked:  output/twoBulk_multiscref_simu_G1000_K4.RData
    Untracked:  output/twoBulk_null_bulkref_simu_G1000_K4_refls30.RData
    Untracked:  output/twoBulk_null_multiscref_simu_G1000_K4.RData
    Untracked:  output/twoBulk_null_scref_simu_G1000_K4.RData
    Untracked:  output/twoBulk_scref_simu_G1000_K4.RData

Unstaged changes:
    Modified:   analysis/cellsize.Rmd
    Modified:   analysis/cellsize2.Rmd
    Modified:   analysis/mulref_simu.Rmd
    Modified:   analysis/multiref_random_nondiag.Rmd
    Modified:   analysis/real_data_simu.Rmd
    Modified:   code/deconference_estfunc.R
    Modified:   code/deconference_main.R
    Modified:   code/deconference_meta.R
    Modified:   code/deconference_multiref.R
    Modified:   code/deconference_setdata.R
    Modified:   code/deconference_summary.R
    Modified:   code/simu_func_multibulk.R
    Modified:   code/unadjusted_lm.R
    Deleted:    data/pbmc.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/music_genelength.Rmd</code>) and HTML (<code>docs/music_genelength.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/e366a21db9321aa742c44a6a6f929648c9d09b64/analysis/music_genelength.Rmd" target="_blank">e366a21</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-10-29
</td>
<td>
wflow_publish(“analysis/music_genelength.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DongyueXie/deconference/4a38e403dbd09d2188799ebe5b84c64cf17189ee/docs/music_genelength.html" target="_blank">4a38e40</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-10-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/1d9a517ac50f9bb7fc7ac777fad53753ab012811/analysis/music_genelength.Rmd" target="_blank">1d9a517</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-10-29
</td>
<td>
wflow_publish(“analysis/music_genelength.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DongyueXie/deconference/2d30de45993bdc419f620d27de04eef55c5bf81f/docs/music_genelength.html" target="_blank">2d30de4</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-10-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DongyueXie/deconference/blob/966330d838128387e987ab7fc915b2a9234e1a38/analysis/music_genelength.Rmd" target="_blank">966330d</a>
</td>
<td>
DongyueXie
</td>
<td>
2020-10-29
</td>
<td>
wflow_publish(“analysis/music_genelength.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="simulation" class="section level1">
<h1>Simulation</h1>
<p>Merge a non-UMI scRNA-seq dataset as bulk data, then use two UMI scRNA-seq dataset as reference data.</p>
<p>Compare input a. gene-length adjusted bulk; b. raw bulk</p>
<pre class="r"><code>library(MuSiC)
library(Biobase)
library(xbioc)
source(&#39;code/deconference_main.R&#39;)

# 4
baron &lt;- readRDS(&quot;data/pancreas/baron.rds&quot;)

# 4
muraro &lt;- readRDS(&quot;data/pancreas/muraro.rds&quot;)

# 10
seger &lt;- readRDS(&quot;data/pancreas/segerstolpe_raw.rds&quot;)

# 8
#enge = readRDS(&quot;data/pancreas/enge_raw.rds&quot;)

# 18
#xin = readRDS(&#39;data/pancreas/xin_raw.rds&#39;)
gene_length = readRDS(&#39;data/gene_length.rds&#39;)</code></pre>
<pre class="r"><code>create_bulk = function(data.obj,indis,cell_types,min_prop=0.15){
  G = nrow(data.obj)
  bulk_counts = matrix(nrow=G,ncol=length(indis))
  col_condition = c()
  bulk_prop = c()
  bulk_ncell = c()
  #bulk_prop = matrix(nrow=length(indis),ncol = length(cell_types))
  for(i in 1:length(indis)){
    idx.temp = which(data.obj$individual==indis[i]&amp;data.obj$cell_type%in%cell_types)
    cell_counts = table(data.obj$cell_type[idx.temp])
    
    if(!is.null(min_prop)){
      
      min_count = min(cell_counts)
    max_cell = floor(min(cell_counts)/min_prop)
    if(max_cell&lt;sum(cell_counts)){
      
      to_rm = sum(cell_counts) - max_cell
      for(j in 1:length(cell_counts)){
        rm.celltype = names(cell_counts)[which.max(cell_counts)]
        candidate.rm.idx = which(data.obj[,idx.temp]$cell_type==rm.celltype)
        if(to_rm&gt;(max(cell_counts)-min_count)){
          idx.temp = idx.temp[-candidate.rm.idx[sample(1:length(candidate.rm.idx),max(cell_counts)-min_count)]]
          to_rm = to_rm - max(cell_counts) + min_count
          cell_counts = table(data.obj$cell_type[idx.temp])
        }else{
          idx.temp = idx.temp[-candidate.rm.idx[sample(1:length(candidate.rm.idx),to_rm)]]
          cell_counts = table(data.obj$cell_type[idx.temp])
          break
        }
        
      }
      # while (to_rm&gt;0&amp;max(cell_counts)&gt;(max_cell*min_prop)) {
      #   rm.celltype = names(cell_counts)[which.max(cell_counts)]
      #   candidate.rm.idx = which(data.obj[,idx.temp]$cell_type==rm.celltype)
      #   idx.temp = idx.temp[-candidate.rm.idx[sample(1:length(candidate.rm.idx),1)]]
      #   cell_counts = table(data.obj$cell_type[idx.temp])
      #   to_rm = to_rm - 1
      # }
    }
    
    }
    
    
    cell_prop = cell_counts/sum(cell_counts)
    bulk_prop = rbind(bulk_prop,cell_prop)
    bulk_ncell = rbind(bulk_ncell,cell_counts)
    bulk_counts[,i] = rowSums(counts(data.obj)[,idx.temp])
    #col_condition[i] = (data.obj$disease[which(data.obj$individual==indis[i])])[1]
    # bulk_prop[i,] = 
  }
  colnames(bulk_counts) = indis
  rownames(bulk_counts) = rownames(data.obj)
  rownames(bulk_ncell) = indis
  rownames(bulk_prop) = indis
  #ii = which(data.obj$individual%in%indis&amp;data.obj$cell_type%in%cell_types)
  #bulk_prop = table(data.obj$individual[ii],data.obj$cell_type[ii])
  #bulk_prop = bulk_prop/rowSums(bulk_prop)
  coldata = DataFrame(individual = indis)
  #browser()
  out = SingleCellExperiment(assays = list(counts = as.matrix(bulk_counts)),
                                   colData = coldata)
  return(list(bulk = out,bulk_prop = bulk_prop,bulk_ncell=bulk_ncell))
}


sce_to_es = function(sce){
  #bulk.data = ExpressionSet(assayData = counts(enge_bulk$bulk))
  pheno_Data = data.frame(cell_type = sce$cell_type,individual = sce$individual)
  rownames(pheno_Data) = colnames(sce)
  ref.data = ExpressionSet(assayData = counts(sce),phenoData = AnnotatedDataFrame(pheno_Data))
  ref.data
}</code></pre>
<div id="seger" class="section level2">
<h2>seger</h2>
<pre class="r"><code>cell_types = c(&#39;acinar&#39;,&#39;alpha&#39;,&#39;beta&#39;,&#39;delta&#39;,&#39;ductal&#39;,&#39;gamma&#39;)
bulk.obj = create_bulk(seger,indis=unique(seger$individual),cell_types = cell_types,min_prop = NULL)</code></pre>
<p>adjust gene length</p>
<pre class="r"><code>refs = set_data_multiref(list(baron=baron,muraro=muraro),bulk.obj$bulk,gene_length = gene_length,protocols = c(&#39;UMI&#39;,&#39;UMI&#39;),cell_types = cell_types)
baron2 = refs$ref.obj$baron
muraro2 = refs$ref.obj$muraro
colData(baron2) = DataFrame(cell_type=baron2$cell_type,individual=baron2$individual)
colData(muraro2) = DataFrame(cell_type=muraro2$cell_type,individual=muraro2$individual)
ref2 = cbind(baron2,muraro2)

m_bulk = ExpressionSet(assayData = counts(refs$bulk.obj))
m_ref2 = sce_to_es(ref2)
music_out2 = music_prop(m_bulk,m_ref2,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;)</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 9143 common genes...</code></pre>
<pre><code>Used 6 cell types in deconvolution...</code></pre>
<pre><code>43 has common genes 9059 ...</code></pre>
<pre><code>25 has common genes 9126 ...</code></pre>
<pre><code>57 has common genes 9129 ...</code></pre>
<pre><code>23 has common genes 9102 ...</code></pre>
<pre><code>48 has common genes 9123 ...</code></pre>
<pre><code>22 has common genes 9121 ...</code></pre>
<pre><code>37 has common genes 9136 ...</code></pre>
<pre><code>27 has common genes 9106 ...</code></pre>
<pre><code>52 has common genes 9122 ...</code></pre>
<pre><code>55 has common genes 9124 ...</code></pre>
<pre class="r"><code>music_out2_s = music_prop(m_bulk,m_ref2,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;,cell_size = data.frame(cell_types=cell_types,cell_size=rep(1,length(cell_types))))</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 9143 common genes...</code></pre>
<pre><code>Used 6 cell types in deconvolution...</code></pre>
<pre><code>43 has common genes 9059 ...</code></pre>
<pre><code>25 has common genes 9126 ...</code></pre>
<pre><code>57 has common genes 9129 ...</code></pre>
<pre><code>23 has common genes 9102 ...</code></pre>
<pre><code>48 has common genes 9123 ...</code></pre>
<pre><code>22 has common genes 9121 ...</code></pre>
<pre><code>37 has common genes 9136 ...</code></pre>
<pre><code>27 has common genes 9106 ...</code></pre>
<pre><code>52 has common genes 9122 ...</code></pre>
<pre><code>55 has common genes 9124 ...</code></pre>
<p>do not adjust gene length</p>
<pre class="r"><code># make sure use the same set of genes.
gene_length2 = gene_length
gene_length2[gene_length2!=1] = 1
refs = set_data_multiref(list(baron=baron,muraro=muraro),bulk.obj$bulk,gene_length = gene_length2,protocols = c(&#39;UMI&#39;,&#39;UMI&#39;),cell_types = cell_types)</code></pre>
<pre class="r"><code>baron2 = refs$ref.obj$baron
muraro2 = refs$ref.obj$muraro
colData(baron2) = DataFrame(cell_type=baron2$cell_type,individual=baron2$individual)
colData(muraro2) = DataFrame(cell_type=muraro2$cell_type,individual=muraro2$individual)
ref2 = cbind(baron2,muraro2)

# run music
ref1 = ref2
m_bulk = ExpressionSet(assayData = counts(refs$bulk.obj))
m_ref1 = sce_to_es(ref1)
music_out1 = music_prop(m_bulk,m_ref1,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;)</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 9143 common genes...</code></pre>
<pre><code>Used 6 cell types in deconvolution...</code></pre>
<pre><code>43 has common genes 9059 ...</code></pre>
<pre><code>25 has common genes 9126 ...</code></pre>
<pre><code>57 has common genes 9129 ...</code></pre>
<pre><code>23 has common genes 9102 ...</code></pre>
<pre><code>48 has common genes 9123 ...</code></pre>
<pre><code>22 has common genes 9121 ...</code></pre>
<pre><code>37 has common genes 9136 ...</code></pre>
<pre><code>27 has common genes 9106 ...</code></pre>
<pre><code>52 has common genes 9122 ...</code></pre>
<pre><code>55 has common genes 9124 ...</code></pre>
<pre class="r"><code>music_out1_s = music_prop(m_bulk,m_ref1,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;,cell_size = data.frame(cell_types=cell_types,cell_size=rep(1,length(cell_types))))</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 9143 common genes...</code></pre>
<pre><code>Used 6 cell types in deconvolution...</code></pre>
<pre><code>43 has common genes 9059 ...</code></pre>
<pre><code>25 has common genes 9126 ...</code></pre>
<pre><code>57 has common genes 9129 ...</code></pre>
<pre><code>23 has common genes 9102 ...</code></pre>
<pre><code>48 has common genes 9123 ...</code></pre>
<pre><code>22 has common genes 9121 ...</code></pre>
<pre><code>37 has common genes 9136 ...</code></pre>
<pre><code>27 has common genes 9106 ...</code></pre>
<pre><code>52 has common genes 9122 ...</code></pre>
<pre><code>55 has common genes 9124 ...</code></pre>
<pre class="r"><code>Eval_multi(prop.real = data.matrix(bulk.obj$bulk_prop), 
           prop.est = list(data.matrix(music_out1$Est.prop.weighted), 
                           data.matrix(music_out2$Est.prop.weighted),
                           data.matrix(music_out1_s$Est.prop.weighted), 
                           data.matrix(music_out2_s$Est.prop.weighted)),
           method.name = c(&#39;raw&#39;,&#39;gene-len-adj&#39;,&#39;raw-fix S&#39;,&#39;gene-len-adj-fix S&#39;))</code></pre>
<pre><code>                      RMSD     mAD      R
raw                0.09535 0.06872 0.9404
gene-len-adj       0.15885 0.11768 0.8729
raw-fix S          0.09257 0.06932 0.9344
gene-len-adj-fix S 0.14545 0.11231 0.8714</code></pre>
</div>
<div id="xin" class="section level2">
<h2>Xin</h2>
<pre class="r"><code>xin &lt;- readRDS(&quot;data/pancreas/xin_raw.rds&quot;)</code></pre>
<pre class="r"><code>cell_types = c(&#39;alpha&#39;,&#39;beta&#39;,&#39;delta&#39;,&#39;gamma&#39;)
bulk.obj = create_bulk(xin,indis=unique(xin$individual),cell_types = cell_types,min_prop = NULL)</code></pre>
<p>adjust gene length</p>
<pre class="r"><code>refs = set_data_multiref(list(baron=baron,muraro=muraro),bulk.obj$bulk,gene_length = gene_length,protocols = c(&#39;UMI&#39;,&#39;UMI&#39;),cell_types = cell_types)
baron2 = refs$ref.obj$baron
muraro2 = refs$ref.obj$muraro
colData(baron2) = DataFrame(cell_type=baron2$cell_type,individual=baron2$individual)
colData(muraro2) = DataFrame(cell_type=muraro2$cell_type,individual=muraro2$individual)
ref2 = cbind(baron2,muraro2)

m_bulk = ExpressionSet(assayData = counts(refs$bulk.obj))
m_ref2 = sce_to_es(ref2)
music_out2 = music_prop(m_bulk,m_ref2,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;)</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8832 common genes...</code></pre>
<pre><code>Used 4 cell types in deconvolution...</code></pre>
<pre><code>Non T2D 1 has common genes 8758 ...</code></pre>
<pre><code>Non T2D 2 has common genes 8410 ...</code></pre>
<pre><code>Non T2D 3 has common genes 8585 ...</code></pre>
<pre><code>Non T2D 5 has common genes 8730 ...</code></pre>
<pre><code>Non T2D 6 has common genes 8663 ...</code></pre>
<pre><code>Non T2D 7 has common genes 8552 ...</code></pre>
<pre><code>Non T2D 8 has common genes 8644 ...</code></pre>
<pre><code>Non T2D 9 has common genes 8612 ...</code></pre>
<pre><code>Non T2D 10 has common genes 8658 ...</code></pre>
<pre><code>Non T2D 11 has common genes 8695 ...</code></pre>
<pre><code>Non T2D 12 has common genes 8791 ...</code></pre>
<pre><code>Non T2D 4 has common genes 8588 ...</code></pre>
<pre><code>T2D 2 has common genes 8745 ...</code></pre>
<pre><code>T2D 3 has common genes 8669 ...</code></pre>
<pre><code>T2D 1 has common genes 8756 ...</code></pre>
<pre><code>T2D 4 has common genes 8808 ...</code></pre>
<pre><code>T2D 5 has common genes 8788 ...</code></pre>
<pre><code>T2D 6 has common genes 8791 ...</code></pre>
<pre class="r"><code>music_out2_s = music_prop(m_bulk,m_ref2,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;,cell_size = data.frame(cell_types=cell_types,cell_size=rep(1,length(cell_types))))</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8832 common genes...</code></pre>
<pre><code>Used 4 cell types in deconvolution...</code></pre>
<pre><code>Non T2D 1 has common genes 8758 ...</code></pre>
<pre><code>Non T2D 2 has common genes 8410 ...</code></pre>
<pre><code>Non T2D 3 has common genes 8585 ...</code></pre>
<pre><code>Non T2D 5 has common genes 8730 ...</code></pre>
<pre><code>Non T2D 6 has common genes 8663 ...</code></pre>
<pre><code>Non T2D 7 has common genes 8552 ...</code></pre>
<pre><code>Non T2D 8 has common genes 8644 ...</code></pre>
<pre><code>Non T2D 9 has common genes 8612 ...</code></pre>
<pre><code>Non T2D 10 has common genes 8658 ...</code></pre>
<pre><code>Non T2D 11 has common genes 8695 ...</code></pre>
<pre><code>Non T2D 12 has common genes 8791 ...</code></pre>
<pre><code>Non T2D 4 has common genes 8588 ...</code></pre>
<pre><code>T2D 2 has common genes 8745 ...</code></pre>
<pre><code>T2D 3 has common genes 8669 ...</code></pre>
<pre><code>T2D 1 has common genes 8756 ...</code></pre>
<pre><code>T2D 4 has common genes 8808 ...</code></pre>
<pre><code>T2D 5 has common genes 8788 ...</code></pre>
<pre><code>T2D 6 has common genes 8791 ...</code></pre>
<p>do not adjust gene length</p>
<pre class="r"><code>refs = set_data_multiref(list(baron=baron,muraro=muraro),bulk.obj$bulk,gene_length = gene_length2,protocols = c(&#39;UMI&#39;,&#39;UMI&#39;),cell_types = cell_types)</code></pre>
<pre class="r"><code>baron2 = refs$ref.obj$baron
muraro2 = refs$ref.obj$muraro
colData(baron2) = DataFrame(cell_type=baron2$cell_type,individual=baron2$individual)
colData(muraro2) = DataFrame(cell_type=muraro2$cell_type,individual=muraro2$individual)
ref2 = cbind(baron2,muraro2)

# run music
ref1 = ref2
m_bulk = ExpressionSet(assayData = counts(refs$bulk.obj))
m_ref1 = sce_to_es(ref1)
music_out1 = music_prop(m_bulk,m_ref1,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;)</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8832 common genes...</code></pre>
<pre><code>Used 4 cell types in deconvolution...</code></pre>
<pre><code>Non T2D 1 has common genes 8758 ...</code></pre>
<pre><code>Non T2D 2 has common genes 8410 ...</code></pre>
<pre><code>Non T2D 3 has common genes 8585 ...</code></pre>
<pre><code>Non T2D 5 has common genes 8730 ...</code></pre>
<pre><code>Non T2D 6 has common genes 8663 ...</code></pre>
<pre><code>Non T2D 7 has common genes 8552 ...</code></pre>
<pre><code>Non T2D 8 has common genes 8644 ...</code></pre>
<pre><code>Non T2D 9 has common genes 8612 ...</code></pre>
<pre><code>Non T2D 10 has common genes 8658 ...</code></pre>
<pre><code>Non T2D 11 has common genes 8695 ...</code></pre>
<pre><code>Non T2D 12 has common genes 8791 ...</code></pre>
<pre><code>Non T2D 4 has common genes 8588 ...</code></pre>
<pre><code>T2D 2 has common genes 8745 ...</code></pre>
<pre><code>T2D 3 has common genes 8669 ...</code></pre>
<pre><code>T2D 1 has common genes 8756 ...</code></pre>
<pre><code>T2D 4 has common genes 8808 ...</code></pre>
<pre><code>T2D 5 has common genes 8788 ...</code></pre>
<pre><code>T2D 6 has common genes 8791 ...</code></pre>
<pre class="r"><code>music_out1_s = music_prop(m_bulk,m_ref1,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;,cell_size = data.frame(cell_types=cell_types,cell_size=rep(1,length(cell_types))))</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8832 common genes...</code></pre>
<pre><code>Used 4 cell types in deconvolution...</code></pre>
<pre><code>Non T2D 1 has common genes 8758 ...</code></pre>
<pre><code>Non T2D 2 has common genes 8410 ...</code></pre>
<pre><code>Non T2D 3 has common genes 8585 ...</code></pre>
<pre><code>Non T2D 5 has common genes 8730 ...</code></pre>
<pre><code>Non T2D 6 has common genes 8663 ...</code></pre>
<pre><code>Non T2D 7 has common genes 8552 ...</code></pre>
<pre><code>Non T2D 8 has common genes 8644 ...</code></pre>
<pre><code>Non T2D 9 has common genes 8612 ...</code></pre>
<pre><code>Non T2D 10 has common genes 8658 ...</code></pre>
<pre><code>Non T2D 11 has common genes 8695 ...</code></pre>
<pre><code>Non T2D 12 has common genes 8791 ...</code></pre>
<pre><code>Non T2D 4 has common genes 8588 ...</code></pre>
<pre><code>T2D 2 has common genes 8745 ...</code></pre>
<pre><code>T2D 3 has common genes 8669 ...</code></pre>
<pre><code>T2D 1 has common genes 8756 ...</code></pre>
<pre><code>T2D 4 has common genes 8808 ...</code></pre>
<pre><code>T2D 5 has common genes 8788 ...</code></pre>
<pre><code>T2D 6 has common genes 8791 ...</code></pre>
<pre class="r"><code>Eval_multi(prop.real = data.matrix(bulk.obj$bulk_prop), 
           prop.est = list(data.matrix(music_out1$Est.prop.weighted), 
                           data.matrix(music_out2$Est.prop.weighted),
                           data.matrix(music_out1_s$Est.prop.weighted), 
                           data.matrix(music_out2_s$Est.prop.weighted)),
           method.name = c(&#39;raw&#39;,&#39;gene-len-adj&#39;,&#39;raw-fix S&#39;,&#39;gene-len-adj-fix S&#39;))</code></pre>
<pre><code>                      RMSD     mAD      R
raw                0.16296 0.13693 0.7441
gene-len-adj       0.12543 0.09872 0.8955
raw-fix S          0.19008 0.15885 0.6737
gene-len-adj-fix S 0.10299 0.07426 0.9275</code></pre>
</div>
<div id="enge" class="section level2">
<h2>enge</h2>
<pre class="r"><code>enge &lt;- readRDS(&quot;data/pancreas/enge_raw.rds&quot;)</code></pre>
<pre class="r"><code>cell_types = c(&#39;acinar&#39;,&#39;alpha&#39;,&#39;beta&#39;,&#39;delta&#39;,&#39;ductal&#39;)
bulk.obj = create_bulk(enge,indis=unique(enge$individual)[-6],cell_types = cell_types,min_prop = NULL)</code></pre>
<p>adjust gene length</p>
<pre class="r"><code>refs = set_data_multiref(list(baron=baron,muraro=muraro),bulk.obj$bulk,gene_length = gene_length,protocols = c(&#39;UMI&#39;,&#39;UMI&#39;),cell_types = cell_types)
baron2 = refs$ref.obj$baron
muraro2 = refs$ref.obj$muraro
colData(baron2) = DataFrame(cell_type=baron2$cell_type,individual=baron2$individual)
colData(muraro2) = DataFrame(cell_type=muraro2$cell_type,individual=muraro2$individual)
ref2 = cbind(baron2,muraro2)

m_bulk = ExpressionSet(assayData = counts(refs$bulk.obj))
m_ref2 = sce_to_es(ref2)
music_out2 = music_prop(m_bulk,m_ref2,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;)</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8964 common genes...</code></pre>
<pre><code>Used 5 cell types in deconvolution...</code></pre>
<pre><code>donor_age: 21 has common genes 8950 ...</code></pre>
<pre><code>donor_age: 38 has common genes 8960 ...</code></pre>
<pre><code>donor_age: 1 has common genes 8861 ...</code></pre>
<pre><code>donor_age: 5 has common genes 8947 ...</code></pre>
<pre><code>donor_age: 44 has common genes 8949 ...</code></pre>
<pre><code>donor_age: 54 has common genes 8935 ...</code></pre>
<pre><code>donor_age: 22 has common genes 8928 ...</code></pre>
<pre class="r"><code>music_out2_s = music_prop(m_bulk,m_ref2,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;,cell_size = data.frame(cell_types=cell_types,cell_size=rep(1,length(cell_types))))</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8964 common genes...</code></pre>
<pre><code>Used 5 cell types in deconvolution...</code></pre>
<pre><code>donor_age: 21 has common genes 8950 ...</code></pre>
<pre><code>donor_age: 38 has common genes 8960 ...</code></pre>
<pre><code>donor_age: 1 has common genes 8861 ...</code></pre>
<pre><code>donor_age: 5 has common genes 8947 ...</code></pre>
<pre><code>donor_age: 44 has common genes 8949 ...</code></pre>
<pre><code>donor_age: 54 has common genes 8935 ...</code></pre>
<pre><code>donor_age: 22 has common genes 8928 ...</code></pre>
<p>do not adjust gene length</p>
<pre class="r"><code>refs = set_data_multiref(list(baron=baron,muraro=muraro),bulk.obj$bulk,gene_length = gene_length2,protocols = c(&#39;UMI&#39;,&#39;UMI&#39;),cell_types = cell_types)</code></pre>
<pre class="r"><code>baron2 = refs$ref.obj$baron
muraro2 = refs$ref.obj$muraro
colData(baron2) = DataFrame(cell_type=baron2$cell_type,individual=baron2$individual)
colData(muraro2) = DataFrame(cell_type=muraro2$cell_type,individual=muraro2$individual)
ref2 = cbind(baron2,muraro2)

# run music
ref1 = ref2
m_bulk = ExpressionSet(assayData = counts(refs$bulk.obj))
m_ref1 = sce_to_es(ref1)
music_out1 = music_prop(m_bulk,m_ref1,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;)</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8964 common genes...</code></pre>
<pre><code>Used 5 cell types in deconvolution...</code></pre>
<pre><code>donor_age: 21 has common genes 8950 ...</code></pre>
<pre><code>donor_age: 38 has common genes 8960 ...</code></pre>
<pre><code>donor_age: 1 has common genes 8861 ...</code></pre>
<pre><code>donor_age: 5 has common genes 8947 ...</code></pre>
<pre><code>donor_age: 44 has common genes 8949 ...</code></pre>
<pre><code>donor_age: 54 has common genes 8935 ...</code></pre>
<pre><code>donor_age: 22 has common genes 8928 ...</code></pre>
<pre class="r"><code>music_out1_s = music_prop(m_bulk,m_ref1,clusters = &#39;cell_type&#39;,samples = &#39;individual&#39;,cell_size = data.frame(cell_types=cell_types,cell_size=rep(1,length(cell_types))))</code></pre>
<pre><code>Creating Relative Abudance Matrix...</code></pre>
<pre><code>Creating Variance Matrix...</code></pre>
<pre><code>Creating Library Size Matrix...</code></pre>
<pre><code>Used 8964 common genes...</code></pre>
<pre><code>Used 5 cell types in deconvolution...</code></pre>
<pre><code>donor_age: 21 has common genes 8950 ...</code></pre>
<pre><code>donor_age: 38 has common genes 8960 ...</code></pre>
<pre><code>donor_age: 1 has common genes 8861 ...</code></pre>
<pre><code>donor_age: 5 has common genes 8947 ...</code></pre>
<pre><code>donor_age: 44 has common genes 8949 ...</code></pre>
<pre><code>donor_age: 54 has common genes 8935 ...</code></pre>
<pre><code>donor_age: 22 has common genes 8928 ...</code></pre>
<pre class="r"><code>Eval_multi(prop.real = data.matrix(bulk.obj$bulk_prop), 
           prop.est = list(data.matrix(music_out1$Est.prop.weighted), 
                           data.matrix(music_out2$Est.prop.weighted),
                           data.matrix(music_out1_s$Est.prop.weighted), 
                           data.matrix(music_out2_s$Est.prop.weighted)),
           method.name = c(&#39;raw&#39;,&#39;gene-len-adj&#39;,&#39;raw-fix S&#39;,&#39;gene-len-adj-fix S&#39;))</code></pre>
<pre><code>                      RMSD     mAD      R
raw                0.15370 0.11890 0.6585
gene-len-adj       0.15276 0.11223 0.8238
raw-fix S          0.15262 0.11428 0.6515
gene-len-adj-fix S 0.14158 0.10185 0.8155</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] SingleCellExperiment_1.6.0  SummarizedExperiment_1.14.1
 [3] DelayedArray_0.10.0         BiocParallel_1.18.0        
 [5] matrixStats_0.55.0          GenomicRanges_1.36.0       
 [7] GenomeInfoDb_1.20.0         xbioc_0.1.19               
 [9] AnnotationDbi_1.46.0        IRanges_2.18.1             
[11] S4Vectors_0.22.1            Biobase_2.44.0             
[13] BiocGenerics_0.30.0         MuSiC_0.1.1                
[15] ggplot2_3.2.1               nnls_1.4                   

loaded via a namespace (and not attached):
 [1] bit64_0.9-7            assertthat_0.2.1       BiocManager_1.30.4    
 [4] blob_1.2.0             GenomeInfoDbData_1.2.1 yaml_2.2.0            
 [7] pillar_1.4.2           RSQLite_2.1.1          backports_1.1.4       
[10] lattice_0.20-38        quantreg_5.41          glue_1.3.1            
[13] digest_0.6.20          promises_1.0.1         XVector_0.24.0        
[16] checkmate_1.9.4        colorspace_1.4-1       htmltools_0.3.6       
[19] httpuv_1.5.1           Matrix_1.2-18          plyr_1.8.4            
[22] pkgconfig_2.0.2        bibtex_0.4.2           SparseM_1.77          
[25] zlibbioc_1.30.0        purrr_0.3.4            xtable_1.8-4          
[28] scales_1.1.0           whisker_0.3-2          later_0.8.0           
[31] MatrixModels_0.4-1     git2r_0.26.1           tibble_2.1.3          
[34] pkgmaker_0.27          withr_2.1.2            lazyeval_0.2.2        
[37] magrittr_1.5           crayon_1.3.4           memoise_1.1.0         
[40] mcmc_0.9-6             evaluate_0.14          fs_1.3.1              
[43] MASS_7.3-51.4          tools_3.6.1            registry_0.5-1        
[46] lifecycle_0.1.0        stringr_1.4.0          MCMCpack_1.4-5        
[49] munsell_0.5.0          compiler_3.6.1         rlang_0.4.6           
[52] grid_3.6.1             RCurl_1.98-1.1         bitops_1.0-6          
[55] rmarkdown_1.13         gtable_0.3.0           codetools_0.2-16      
[58] DBI_1.1.0              reshape2_1.4.3         R6_2.4.0              
[61] knitr_1.23             dplyr_0.8.3            bit_1.1-14            
[64] workflowr_1.6.2        rprojroot_1.3-2        stringi_1.4.3         
[67] Rcpp_1.0.3             vctrs_0.3.1            tidyselect_1.1.0      
[70] xfun_0.8               coda_0.19-3           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
